# Table of Contents
 <p><div class="lev2 toc-item"><a href="#Geometric" data-toc-modified-id="Geometric-01"><span class="toc-item-num">0.1&nbsp;&nbsp;</span>Geometric</a></div><div class="lev2 toc-item"><a href="#Binomial" data-toc-modified-id="Binomial-02"><span class="toc-item-num">0.2&nbsp;&nbsp;</span>Binomial</a></div><div class="lev2 toc-item"><a href="#Negative-Binomial" data-toc-modified-id="Negative-Binomial-03"><span class="toc-item-num">0.3&nbsp;&nbsp;</span>Negative Binomial</a></div><div class="lev1 toc-item"><a href="#Exercises" data-toc-modified-id="Exercises-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Exercises</a></div><div class="lev3 toc-item"><a href="#2.-Recall-the-committee-example-in-Section-3.9.2.-Suppose-now,-though,-that-the-selection-protocol-is-that-there-must-be-at-least-one-man-and-at-least-one-woman-on-the-committee.-Find-E(D)-and-Var(D)." data-toc-modified-id="2.-Recall-the-committee-example-in-Section-3.9.2.-Suppose-now,-though,-that-the-selection-protocol-is-that-there-must-be-at-least-one-man-and-at-least-one-woman-on-the-committee.-Find-E(D)-and-Var(D).-101"><span class="toc-item-num">1.0.1&nbsp;&nbsp;</span>2. Recall the committee example in Section 3.9.2. Suppose now, though, that the selection protocol is that there must be at least one man and at least one woman on the committee. Find $E(D)$ and $Var(D)$.</a></div><div class="lev3 toc-item"><a href="#3.-Suppose-a-bit-stream-is-subject-to-errors,-with-each-bit-having-probability-p-of-error,-and-with-the-bits-being-independent.-Consider-a-set-of-four-particular-bits.-Let-X-denote-the-number-of-erroneous-bits-among-those-four." data-toc-modified-id="3.-Suppose-a-bit-stream-is-subject-to-errors,-with-each-bit-having-probability-p-of-error,-and-with-the-bits-being-independent.-Consider-a-set-of-four-particular-bits.-Let-X-denote-the-number-of-erroneous-bits-among-those-four.-102"><span class="toc-item-num">1.0.2&nbsp;&nbsp;</span>3. Suppose a bit stream is subject to errors, with each bit having probability p of error, and with the bits being independent. Consider a set of four particular bits. Let $X$ denote the number of erroneous bits among those four.</a></div><div class="lev4 toc-item"><a href="#(a)-Find-P(X-=-2)-and-EX." data-toc-modified-id="(a)-Find-P(X-=-2)-and-EX.-1021"><span class="toc-item-num">1.0.2.1&nbsp;&nbsp;</span>(a) Find $P(X = 2)$ and $EX$.</a></div><div class="lev4 toc-item"><a href="#(b)-What-famous-parametric-family-of-distributions-does-the-distribution-of-X-belong-to?" data-toc-modified-id="(b)-What-famous-parametric-family-of-distributions-does-the-distribution-of-X-belong-to?-1022"><span class="toc-item-num">1.0.2.2&nbsp;&nbsp;</span>(b) What famous parametric family of distributions does the distribution of X belong to?</a></div><div class="lev4 toc-item"><a href="#(c)-Let-Y-denote-the-maximum-number-of-consecutive-erroneous-bits.-Find-P(Y-=-2)-and-Var(Y)." data-toc-modified-id="(c)-Let-Y-denote-the-maximum-number-of-consecutive-erroneous-bits.-Find-P(Y-=-2)-and-Var(Y).-1023"><span class="toc-item-num">1.0.2.3&nbsp;&nbsp;</span>(c) Let $Y$ denote the maximum number of consecutive erroneous bits. Find $P(Y = 2)$ and $Var(Y)$.</a></div><div class="lev3 toc-item"><a href="#4.-Derive-Var(W)-=-frac{1-p}{p^2}." data-toc-modified-id="4.-Derive-Var(W)-=-frac{1-p}{p^2}.-103"><span class="toc-item-num">1.0.3&nbsp;&nbsp;</span>4. Derive $Var(W) = \frac{1-p}{p^2}$.</a></div><div class="lev3 toc-item"><a href="#5.-Finish-the-computation-in-4.18" data-toc-modified-id="5.-Finish-the-computation-in-4.18-104"><span class="toc-item-num">1.0.4&nbsp;&nbsp;</span>5. Finish the computation in 4.18</a></div><div class="lev3 toc-item"><a href="#6.-Derive-the-facts-that-for-a-Poisson-distributed-random-variable-X-with-parameter-λ,-EX-=-Var(X)-=-λ.-Use-the-hints-in-Section-4.5." data-toc-modified-id="6.-Derive-the-facts-that-for-a-Poisson-distributed-random-variable-X-with-parameter-λ,-EX-=-Var(X)-=-λ.-Use-the-hints-in-Section-4.5.-105"><span class="toc-item-num">1.0.5&nbsp;&nbsp;</span>6. Derive the facts that for a Poisson-distributed random variable $X$ with parameter $λ$, $EX = Var(X) = λ$. Use the hints in Section 4.5.</a></div><div class="lev2 toc-item"><a href="#7.-A-civil-engineer-is-collecting-data-on-a-certain-road.-She-needs-to-have-data-on-25-trucks,-and-10-percent-of-the-vehicles-on-that-road-are-trucks.-State-the-famous-parametric-family-that-is-relevant-here,-and-find-the-probability-that-she-will-need-to-wait-for-more-than-200-vehicles-to-pass-before-she-gets-the-needed-data." data-toc-modified-id="7.-A-civil-engineer-is-collecting-data-on-a-certain-road.-She-needs-to-have-data-on-25-trucks,-and-10-percent-of-the-vehicles-on-that-road-are-trucks.-State-the-famous-parametric-family-that-is-relevant-here,-and-find-the-probability-that-she-will-need-to-wait-for-more-than-200-vehicles-to-pass-before-she-gets-the-needed-data.-11"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>7. A civil engineer is collecting data on a certain road. She needs to have data on $25$ trucks, and $10$ percent of the vehicles on that road are trucks. State the famous parametric family that is relevant here, and find the probability that she will need to wait for more than $200$ vehicles to pass before she gets the needed data.</a></div><div class="lev2 toc-item"><a href="#8.-In-the-ALOHA-example:" data-toc-modified-id="8.-In-the-ALOHA-example:-12"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>8. In the ALOHA example:</a></div><div class="lev3 toc-item"><a href="#(a)-Find-E(X_1)-and-Var(X_1),-for-the-case-p-=-0.4,-q-=-0.8.-You-are-welcome-to-use-quantities-already-computed-in-the-text,-e.g.-P(X_1-=-1)-=-0.48,-but-be-sure-to-cite-equation-numbers" data-toc-modified-id="(a)-Find-E(X_1)-and-Var(X_1),-for-the-case-p-=-0.4,-q-=-0.8.-You-are-welcome-to-use-quantities-already-computed-in-the-text,-e.g.-P(X_1-=-1)-=-0.48,-but-be-sure-to-cite-equation-numbers-121"><span class="toc-item-num">1.2.1&nbsp;&nbsp;</span>(a) Find $E(X_1)$ and $Var(X_1)$, for the case $p = 0.4$, $q = 0.8$. You are welcome to use quantities already computed in the text, e.g. $P(X_1 = 1) = 0.48$, but be sure to cite equation numbers</a></div><div class="lev2 toc-item"><a href="#9.-Our-experiment-is-to-toss-a-nickel-until-we-get-a-head,-taking-X-rolls,-and-then-toss-a-dime-until-we-get-a-head,-taking-Y-tosses.-Find:" data-toc-modified-id="9.-Our-experiment-is-to-toss-a-nickel-until-we-get-a-head,-taking-X-rolls,-and-then-toss-a-dime-until-we-get-a-head,-taking-Y-tosses.-Find:-13"><span class="toc-item-num">1.3&nbsp;&nbsp;</span>9. Our experiment is to toss a nickel until we get a head, taking $X$ rolls, and then toss a dime until we get a head, taking $Y$ tosses. Find:</a></div><div class="lev3 toc-item"><a href="#(a)-Var(X+Y)" data-toc-modified-id="(a)-Var(X+Y)-131"><span class="toc-item-num">1.3.1&nbsp;&nbsp;</span>(a) $Var(X+Y)$</a></div><div class="lev3 toc-item"><a href="#(b)-Long-run-average-in-a-“notebook”-column-labeled-X^2." data-toc-modified-id="(b)-Long-run-average-in-a-“notebook”-column-labeled-X^2.-132"><span class="toc-item-num">1.3.2&nbsp;&nbsp;</span>(b) Long-run average in a “notebook” column labeled $X^2$.</a></div><div class="lev2 toc-item"><a href="#10.-Consider-the-game-in-Section-4.8.-Find-E(Z)-and-Var(Z),-where-Z-=-Y_6-−-X_6." data-toc-modified-id="10.-Consider-the-game-in-Section-4.8.-Find-E(Z)-and-Var(Z),-where-Z-=-Y_6-−-X_6.-14"><span class="toc-item-num">1.4&nbsp;&nbsp;</span>10. Consider the game in Section 4.8. Find $E(Z)$ and $Var(Z)$, where $Z = Y_6 − X_6$.</a></div><div class="lev2 toc-item"><a href="#11.-Say-we-choose-six-cards-from-a-standard-deck,-one-at-a-time-WITHOUT-replacement.-Let-N-be-the-number-of-kings-we-get.-Does-N-have-a-binomial-distribution?-Choose-one:-(i)-Yes.-(ii)-No,-since-trials-are-not-independent.-(iii)-No,-since-the-probability-of-success-is-not-constant-from-trial-to-trial.-(iv)-No,-since-the-number-of-trials-is-not-fixed.-(v)-(ii)-and-(iii).-(iv)-(ii)-and-(iv).-(vii)-(iii)-and-(iv)." data-toc-modified-id="11.-Say-we-choose-six-cards-from-a-standard-deck,-one-at-a-time-WITHOUT-replacement.-Let-N-be-the-number-of-kings-we-get.-Does-N-have-a-binomial-distribution?-Choose-one:-(i)-Yes.-(ii)-No,-since-trials-are-not-independent.-(iii)-No,-since-the-probability-of-success-is-not-constant-from-trial-to-trial.-(iv)-No,-since-the-number-of-trials-is-not-fixed.-(v)-(ii)-and-(iii).-(iv)-(ii)-and-(iv).-(vii)-(iii)-and-(iv).-15"><span class="toc-item-num">1.5&nbsp;&nbsp;</span>11. Say we choose six cards from a standard deck, one at a time WITHOUT replacement. Let $N$ be the number of kings we get. Does $N$ have a binomial distribution? Choose one: (i) Yes. (ii) No, since trials are not independent. (iii) No, since the probability of success is not constant from trial to trial. (iv) No, since the number of trials is not fixed. (v) (ii) and (iii). (iv) (ii) and (iv). (vii) (iii) and (iv).</a></div><div class="lev2 toc-item"><a href="#12.-Suppose-we-have-n-independent-trials,-with-the-probability-of-success-on-the-i^{th}-trial-being-p_i.-Let-X-=-text{the-number-of-successes}.-Use-the-fact-that-“the-variance-of-the-sum-is-the-sum-of-the-variance”-for-independent-random-variables-to-derive-Var(X)." data-toc-modified-id="12.-Suppose-we-have-n-independent-trials,-with-the-probability-of-success-on-the-i^{th}-trial-being-p_i.-Let-X-=-text{the-number-of-successes}.-Use-the-fact-that-“the-variance-of-the-sum-is-the-sum-of-the-variance”-for-independent-random-variables-to-derive-Var(X).-16"><span class="toc-item-num">1.6&nbsp;&nbsp;</span>12. Suppose we have $n$ independent trials, with the probability of success on the $i^{th}$ trial being $p_i$. Let $X = \text{the number of successes}$. Use the fact that “the variance of the sum is the sum of the variance” for independent random variables to derive $Var(X)$.</a></div><div class="lev2 toc-item"><a href="#13.-Prove-Equation-3.41" data-toc-modified-id="13.-Prove-Equation-3.41-17"><span class="toc-item-num">1.7&nbsp;&nbsp;</span>13. Prove Equation 3.41</a></div><div class="lev2 toc-item"><a href="#14.-Show-that-if-X-is-a-nonnegative-integer-valued-random-variable,-then" data-toc-modified-id="14.-Show-that-if-X-is-a-nonnegative-integer-valued-random-variable,-then-18"><span class="toc-item-num">1.8&nbsp;&nbsp;</span>14. Show that if X is a nonnegative-integer valued random variable, then</a></div><div class="lev2 toc-item"><a href="#15.-Suppose-we-toss-a-fair-coin-n-times,-resulting-in-X-heads.-Show-that-the-term-expected-value-is-a-misnomer,-by-showing-that" data-toc-modified-id="15.-Suppose-we-toss-a-fair-coin-n-times,-resulting-in-X-heads.-Show-that-the-term-expected-value-is-a-misnomer,-by-showing-that-19"><span class="toc-item-num">1.9&nbsp;&nbsp;</span>15. Suppose we toss a fair coin n times, resulting in X heads. Show that the term expected value is a misnomer, by showing that</a></div><div class="lev2 toc-item"><a href="#Use-Stirling's-approximation," data-toc-modified-id="Use-Stirling's-approximation,-110"><span class="toc-item-num">1.10&nbsp;&nbsp;</span>Use Stirling's approximation,</a></div><div class="lev2 toc-item"><a href="#16.-Suppose-X-and-Y-are-independent-random-variables-with-standard-deviations-3-and-4,-respectively." data-toc-modified-id="16.-Suppose-X-and-Y-are-independent-random-variables-with-standard-deviations-3-and-4,-respectively.-111"><span class="toc-item-num">1.11&nbsp;&nbsp;</span>16. Suppose $X$ and $Y$ are independent random variables with standard deviations $3$ and $4$, respectively.</a></div><div class="lev3 toc-item"><a href="#(a)-Find-Var(X+Y)." data-toc-modified-id="(a)-Find-Var(X+Y).-1111"><span class="toc-item-num">1.11.1&nbsp;&nbsp;</span>(a) Find $Var(X+Y)$.</a></div><div class="lev3 toc-item"><a href="#(b)-Find-Var(2X---Y)." data-toc-modified-id="(b)-Find-Var(2X---Y).-1112"><span class="toc-item-num">1.11.2&nbsp;&nbsp;</span>(b) Find $Var(2X - Y)$.</a></div><div class="lev2 toc-item"><a href="#18.-Let-X-be-the-total-number-of-dots-we-get-if-we-roll-three-dice.-Find-an-upper-bound-for-P(X-geq-15),-using-our-course-materials." data-toc-modified-id="18.-Let-X-be-the-total-number-of-dots-we-get-if-we-roll-three-dice.-Find-an-upper-bound-for-P(X-geq-15),-using-our-course-materials.-112"><span class="toc-item-num">1.12&nbsp;&nbsp;</span>18. Let $X$ be the total number of dots we get if we roll three dice. Find an upper bound for $P(X \geq 15)$, using our course materials.</a></div><div class="lev2 toc-item"><a href="#19.-Suppose-X-and-Y-are-independent-random-variables,-and-let-Z-=-XY.-Show-that-Var(Z)-=-E(X^2)E(Y^2)-−-[E(X)]^2[E(Y-)]^2." data-toc-modified-id="19.-Suppose-X-and-Y-are-independent-random-variables,-and-let-Z-=-XY.-Show-that-Var(Z)-=-E(X^2)E(Y^2)-−-[E(X)]^2[E(Y-)]^2.-113"><span class="toc-item-num">1.13&nbsp;&nbsp;</span>19. Suppose $X$ and $Y$ are independent random variables, and let $Z = XY$. Show that $Var(Z) = E(X^2)E(Y^2) − [E(X)]^2[E(Y )]^2$.</a></div><div class="lev2 toc-item"><a href="#20.-This-problem-involves-a-very-simple-model-of-the-Web.-(Far-more-complex-ones-exist.)" data-toc-modified-id="20.-This-problem-involves-a-very-simple-model-of-the-Web.-(Far-more-complex-ones-exist.)-114"><span class="toc-item-num">1.14&nbsp;&nbsp;</span>20. This problem involves a very simple model of the Web. (Far more complex ones exist.)</a></div><div class="lev2 toc-item"><a href="#Suppose-we-have-n-Websites.-For-each-pair-of-sites-i-and-j,-i-neq-j,-there-is-a-link-from-site-i-to-site-j-with-probability-p,-and-no-link-(in-that-direction)-with-probability-1-p.-Let-N_i-denote-the-number-of-sites-that-site-i-is-linked-to;-note-that-N_i-can-range-from-0-to-n-1.-Also,-let-M_{ij}-denote-the-number-of-outgoing-links-that-sites-i-and-j-have-in-common,-not-counting-the-one-between-them,-if-any.-Assume-that-each-site-forms-its-outgoing-links-independently-of-the-others." data-toc-modified-id="Suppose-we-have-n-Websites.-For-each-pair-of-sites-i-and-j,-i-neq-j,-there-is-a-link-from-site-i-to-site-j-with-probability-p,-and-no-link-(in-that-direction)-with-probability-1-p.-Let-N_i-denote-the-number-of-sites-that-site-i-is-linked-to;-note-that-N_i-can-range-from-0-to-n-1.-Also,-let-M_{ij}-denote-the-number-of-outgoing-links-that-sites-i-and-j-have-in-common,-not-counting-the-one-between-them,-if-any.-Assume-that-each-site-forms-its-outgoing-links-independently-of-the-others.-115"><span class="toc-item-num">1.15&nbsp;&nbsp;</span>Suppose we have $n$ Websites. For each pair of sites $i$ and $j$, $i \neq j$, there is a link from site $i$ to site $j$ with probability $p$, and no link (in that direction) with probability $1-p$. Let $N_i$ denote the number of sites that site $i$ is linked to; note that $N_i$ can range from $0$ to $n-1$. Also, let $M_{ij}$ denote the number of outgoing links that sites $i$ and $j$ have in common, not counting the one between them, if any. Assume that each site forms its outgoing links independently of the others.</a></div><div class="lev3 toc-item"><a href="#(a)-P(N_1-=3)" data-toc-modified-id="(a)-P(N_1-=3)-1151"><span class="toc-item-num">1.15.1&nbsp;&nbsp;</span>(a) $P(N_1 =3)$</a></div><div class="lev3 toc-item"><a href="#(b)-P(N_1-=-3-text{-and-}-N_2-=-2)" data-toc-modified-id="(b)-P(N_1-=-3-text{-and-}-N_2-=-2)-1152"><span class="toc-item-num">1.15.2&nbsp;&nbsp;</span>(b) $P(N_1 = 3 \text{ and } N_2 = 2)$</a></div><div class="lev3 toc-item"><a href="#(c)-Var(N_1)" data-toc-modified-id="(c)-Var(N_1)-1153"><span class="toc-item-num">1.15.3&nbsp;&nbsp;</span>(c) $Var(N_1)$</a></div><div class="lev3 toc-item"><a href="#(d)-Var(N_1-+-N_2)" data-toc-modified-id="(d)-Var(N_1-+-N_2)-1154"><span class="toc-item-num">1.15.4&nbsp;&nbsp;</span>(d) $Var(N_1 + N_2)$</a></div><div class="lev2 toc-item"><a href="#Note:-There-are-some-good-shortcuts-in-some-of-these-problems,-making-the-work-much-easier." data-toc-modified-id="Note:-There-are-some-good-shortcuts-in-some-of-these-problems,-making-the-work-much-easier.-116"><span class="toc-item-num">1.16&nbsp;&nbsp;</span>Note: There are some good shortcuts in some of these problems, making the work much easier.</a></div><div class="lev2 toc-item"><a href="#21.-Let-X-denote-the-number-of-heads-we-get-by-tossing-a-coin-50-times.-Consider-Chebychev’s-Inequality-for-the-case-of-2-standard-deviations.-Compare-the-upper-bound-given-by-the-inequality-to-the-exact-probability." data-toc-modified-id="21.-Let-X-denote-the-number-of-heads-we-get-by-tossing-a-coin-50-times.-Consider-Chebychev’s-Inequality-for-the-case-of-2-standard-deviations.-Compare-the-upper-bound-given-by-the-inequality-to-the-exact-probability.-117"><span class="toc-item-num">1.17&nbsp;&nbsp;</span>21. Let $X$ denote the number of heads we get by tossing a coin $50$ times. Consider Chebychev’s Inequality for the case of $2$ standard deviations. Compare the upper bound given by the inequality to the exact probability.</a></div><div class="lev2 toc-item"><a href="#22.-Suppose-the-number-N-of-cars-arriving-during-a-given-time-period-at-a-toll-booth-has-a-Poisson-distribution-with-parameter-lambda.-Each-car-has-a-probability-p-of-being-in-a-car-pool.-Let-M-be-the-number-of-car-pool-cars-that-arrive-in-the-given-period.-Show-that-M-also-has-a-Poisson-distribution,-with-parameter-p\lambda.-(Hint:-Use-the-Maclaurin-series-for-e^x.)" data-toc-modified-id="22.-Suppose-the-number-N-of-cars-arriving-during-a-given-time-period-at-a-toll-booth-has-a-Poisson-distribution-with-parameter-lambda.-Each-car-has-a-probability-p-of-being-in-a-car-pool.-Let-M-be-the-number-of-car-pool-cars-that-arrive-in-the-given-period.-Show-that-M-also-has-a-Poisson-distribution,-with-parameter-p\lambda.-(Hint:-Use-the-Maclaurin-series-for-e^x.)-118"><span class="toc-item-num">1.18&nbsp;&nbsp;</span>22. Suppose the number $N$ of cars arriving during a given time period at a toll booth has a Poisson distribution with parameter $\lambda$. Each car has a probability $p$ of being in a car pool. Let $M$ be the number of car-pool cars that arrive in the given period. Show that $M$ also has a Poisson distribution, with parameter $p\lambda$. (Hint: Use the Maclaurin series for $e^x$.)</a></div><div class="lev2 toc-item"><a href="#23.-Consider-a-three-sided-die,-as-on-page-36.-Let-X-denote-the-number-of-dots-obtained-in-one-roll." data-toc-modified-id="23.-Consider-a-three-sided-die,-as-on-page-36.-Let-X-denote-the-number-of-dots-obtained-in-one-roll.-119"><span class="toc-item-num">1.19&nbsp;&nbsp;</span>23. Consider a three-sided die, as on page 36. Let X denote the number of dots obtained in one roll.</a></div><div class="lev2 toc-item"><a href="#(a)-State-the-value-of-p_x(2):" data-toc-modified-id="(a)-State-the-value-of-p_x(2):-120"><span class="toc-item-num">1.20&nbsp;&nbsp;</span>(a) State the value of $p_x(2):$</a></div><div class="lev2 toc-item"><a href="#(b)-Find-EX-and-Var(X):" data-toc-modified-id="(b)-Find-EX-and-Var(X):-121"><span class="toc-item-num">1.21&nbsp;&nbsp;</span>(b) Find $EX$ and $Var(X)$:</a></div><div class="lev2 toc-item"><a href="#(c)-Suppose-you-win-2-for-each-dot.-Find-EW,-where-W-is-the-amount-you-win:" data-toc-modified-id="(c)-Suppose-you-win-2-for-each-dot.-Find-EW,-where-W-is-the-amount-you-win:-122"><span class="toc-item-num">1.22&nbsp;&nbsp;</span>(c) Suppose you win $\$2$ for each dot. Find $EW$, where $W$ is the amount you win:</a></div><div class="lev2 toc-item"><a href="#24.-Consider-the-parking-space-problem-in-Section-4.2.2.-Find-Var(M),-where-M-is-the-number-of-empty-spaces-in-the-first-block,-and-Var(D)." data-toc-modified-id="24.-Consider-the-parking-space-problem-in-Section-4.2.2.-Find-Var(M),-where-M-is-the-number-of-empty-spaces-in-the-first-block,-and-Var(D).-123"><span class="toc-item-num">1.23&nbsp;&nbsp;</span>24. Consider the parking space problem in Section 4.2.2. Find $Var(M)$, where $M$ is the number of empty spaces in the first block, and $Var(D)$.</a></div><div class="lev2 toc-item"><a href="#25.-Suppose-X-and-Y-are-independent,-with-variances-1-and-2,-respectively.-Find-the-value-of-c-that-minimizes-Var[cX-+-(1-c)Y]." data-toc-modified-id="25.-Suppose-X-and-Y-are-independent,-with-variances-1-and-2,-respectively.-Find-the-value-of-c-that-minimizes-Var[cX-+-(1-c)Y].-124"><span class="toc-item-num">1.24&nbsp;&nbsp;</span>25. Suppose $X$ and $Y$ are independent, with variances $1$ and $2$, respectively. Find the value of $c$ that minimizes $Var[cX + (1-c)Y]$.</a></div><div class="lev2 toc-item"><a href="#26.-In-the-cards-example-in-Section-2.15.1,-let-H-denote-the-number-of-hearts.-Find-EH-and-Var(H)." data-toc-modified-id="26.-In-the-cards-example-in-Section-2.15.1,-let-H-denote-the-number-of-hearts.-Find-EH-and-Var(H).-125"><span class="toc-item-num">1.25&nbsp;&nbsp;</span>26. In the cards example in Section 2.15.1, let $H$ denote the number of hearts. Find $EH$ and $Var(H)$.</a></div><div class="lev2 toc-item"><a href="#27.-In-the-bank-example-in-Section-4.5,-suppose-you-observe-the-bank-for-n-days.-Let-X-denote-the-number-of-days-in-which-at-least-2-customers-entered-during-the-11:00-11:15-observation-period.-Find-P(X-=-k)." data-toc-modified-id="27.-In-the-bank-example-in-Section-4.5,-suppose-you-observe-the-bank-for-n-days.-Let-X-denote-the-number-of-days-in-which-at-least-2-customers-entered-during-the-11:00-11:15-observation-period.-Find-P(X-=-k).-126"><span class="toc-item-num">1.26&nbsp;&nbsp;</span>27. In the bank example in Section 4.5, suppose you observe the bank for $n$ days. Let $X$ denote the number of days in which at least $2$ customers entered during the 11:00-11:15 observation period. Find $P(X = k)$.</a></div><div class="lev2 toc-item"><a href="#28.-Find-E(X^3),-where-X-has-a-geometric-distribution-with-parameter-p." data-toc-modified-id="28.-Find-E(X^3),-where-X-has-a-geometric-distribution-with-parameter-p.-127"><span class="toc-item-num">1.27&nbsp;&nbsp;</span>28. Find $E(X^3)$, where $X$ has a geometric distribution with parameter $p$.</a></div><div class="lev2 toc-item"><a href="#29.-Supppose-we-have-a-nonnegative-random-variable-X,-and-define-a-new-random-variable-Y,-which-is-equal-to-X-if-X-&gt;-8-and-equal-to-0-otherwise.-Assume-X-takes-on-only-a-finite-number-of-values-(just-a-mathematical-nicety,-not-really-an-issue).-Which-one-of-the-following-is-true:" data-toc-modified-id="29.-Supppose-we-have-a-nonnegative-random-variable-X,-and-define-a-new-random-variable-Y,-which-is-equal-to-X-if-X->-8-and-equal-to-0-otherwise.-Assume-X-takes-on-only-a-finite-number-of-values-(just-a-mathematical-nicety,-not-really-an-issue).-Which-one-of-the-following-is-true:-128"><span class="toc-item-num">1.28&nbsp;&nbsp;</span>29. Supppose we have a nonnegative random variable $X$, and define a new random variable $Y$, which is equal to $X$ if $X &gt; 8$ and equal to $0$ otherwise. Assume $X$ takes on only a finite number of values (just a mathematical nicety, not really an issue). Which one of the following is true:</a></div><div class="lev3 toc-item"><a href="#(i)-EY-leq-EX." data-toc-modified-id="(i)-EY-leq-EX.-1281"><span class="toc-item-num">1.28.1&nbsp;&nbsp;</span>(i) $EY \leq EX$.</a></div><div class="lev3 toc-item"><a href="#(ii)-EY-≥-EX." data-toc-modified-id="(ii)-EY-≥-EX.-1282"><span class="toc-item-num">1.28.2&nbsp;&nbsp;</span>(ii) $EY ≥ EX$.</a></div><div class="lev3 toc-item"><a href="#(iii)-Either-of-EY-and-EX-could-be-larger-than-the-other,-depending-on-the-situation." data-toc-modified-id="(iii)-Either-of-EY-and-EX-could-be-larger-than-the-other,-depending-on-the-situation.-1283"><span class="toc-item-num">1.28.3&nbsp;&nbsp;</span>(iii) Either of $EY$ and $EX$ could be larger than the other, depending on the situation.</a></div><div class="lev3 toc-item"><a href="#(iv)-EY-is-undefined." data-toc-modified-id="(iv)-EY-is-undefined.-1284"><span class="toc-item-num">1.28.4&nbsp;&nbsp;</span>(iv) $EY$ is undefined.</a></div><div class="lev2 toc-item"><a href="#30.-Say-we-roll-two-dice,-a-blue-one-and-a-yellow-one.-Let-B-and-Y-denote-the-number-of-dots-we-get,-respectively,-and-write-S-=-B-+-Y.-Now-let-G-denote-the-indicator-random-variable-for-the-event-S-=-2.-Find-E(G)." data-toc-modified-id="30.-Say-we-roll-two-dice,-a-blue-one-and-a-yellow-one.-Let-B-and-Y-denote-the-number-of-dots-we-get,-respectively,-and-write-S-=-B-+-Y.-Now-let-G-denote-the-indicator-random-variable-for-the-event-S-=-2.-Find-E(G).-129"><span class="toc-item-num">1.29&nbsp;&nbsp;</span>30. Say we roll two dice, a blue one and a yellow one. Let $B$ and $Y$ denote the number of dots we get, respectively, and write $S = B + Y$. Now let $G$ denote the indicator random variable for the event $S = 2$. Find $E(G)$.</a></div><div class="lev2 toc-item"><a href="#32.-Consider-the-bus-ridership-example,-Section-2.12.-Suppose-upon-arrival-to-a-certain-stop,-there-are-2-passengers.-Let-A-denote-the-number-of-them-who-choose-to-alight-at-that-stop." data-toc-modified-id="32.-Consider-the-bus-ridership-example,-Section-2.12.-Suppose-upon-arrival-to-a-certain-stop,-there-are-2-passengers.-Let-A-denote-the-number-of-them-who-choose-to-alight-at-that-stop.-130"><span class="toc-item-num">1.30&nbsp;&nbsp;</span>32. Consider the bus ridership example, Section 2.12. Suppose upon arrival to a certain stop, there are $2$ passengers. Let $A$ denote the number of them who choose to alight at that stop.</a></div><div class="lev3 toc-item"><a href="#33.-Suppose-you-have-a-large-disk-farm,-so-heavily-used-that-the-lifetimes-L-are-measured-in-months.-They-come-from-two-different-factories,-in-proportions-q-and-1-q.-The-disks-from-factory-i-have-geometrically-distributed-lifetime-with-parameter-p_i,-i-=-1,2.-Find-Var(L)-in-terms-of-q-and-the-p_i." data-toc-modified-id="33.-Suppose-you-have-a-large-disk-farm,-so-heavily-used-that-the-lifetimes-L-are-measured-in-months.-They-come-from-two-different-factories,-in-proportions-q-and-1-q.-The-disks-from-factory-i-have-geometrically-distributed-lifetime-with-parameter-p_i,-i-=-1,2.-Find-Var(L)-in-terms-of-q-and-the-p_i.-1301"><span class="toc-item-num">1.30.1&nbsp;&nbsp;</span>33. Suppose you have a large disk farm, so heavily used that the lifetimes $L$ are measured in months. They come from two different factories, in proportions $q$ and $1-q$. The disks from factory $i$ have geometrically distributed lifetime with parameter $p_i$, $i = 1,2$. Find $Var(L)$ in terms of $q$ and the $p_i$.</a></div>

In [1]:
import numpy as np
import scipy.stats as ss
import scipy.misc as ms
import pandas as pd

In [2]:
g = np.random.geometric(p=0.4, size=1000000)

In [3]:
(g==3).sum() / 1000000.0

0.14416100000000001

## Geometric

## Binomial

In [4]:
# Toss coins 5 times. Calculate probability that two of them will be heads.

def coin_coss_5_times_get_2_heads():
    binom = ss.binom(n=5, p=0.5)
    num_heads = 2
    return binom.pmf(num_heads)

coin_coss_5_times_get_2_heads()

0.3125

In [5]:
ms.comb(5,2) * 0.5**2 * 0.5**3

0.3125

## Negative Binomial

In [6]:
# You have a device that uses up battery over several months. 
# You have 3 batteries -- 1 main and 2 spare. 
# Probability that a battery will die in any given month is 0.1.
# Let L be the length of time in months that all 3 would be dead. Find P(L=12)

# nbinom.pmf(k) = choose(k+n-1, n-1) * p**n * (1-p)**k

def three_batt_life():
    num_batteries = 3
    prob_death = 0.1
    nbinom = ss.nbinom(n=num_batteries, p=prob_death)
    num_survives = 9
    return nbinom.pmf(num_survives)

three_batt_life()

0.021308126895000033

In [7]:
ms.comb(11,2) * (0.9**9) * (0.1**3)

0.021308126895000008

# Exercises

1. Consider a game in which one rolls a single die until one accumulates a total of at least four dots. Let X denote the number of rolls needed. Find $P(X \leq 2)$ and $E(X)$.

\begin{equation}
  P(X \leq 2) = P(X = 1) + P(X = 2) \\
\end{equation}

\begin{equation}
  \begin{aligned}
  P(X = 1) &= P(R_1 \geq 4) \\
  &= 1/2 \\
  \end{aligned}
\end{equation}

\begin{equation}
  \begin{aligned}
  P(X = 2) &= P(R_1 = 1)P(R_2 \geq 3) \\
  &\quad + P(R_1 = 2)P(R_2 \geq 2) \\
  &\quad + P(R_1 = 3)P(R_2 \geq 1) \\
  &= 1/6 \times 4/6 \\
  &\quad + 1/6 \times 5/6 \\
  &\quad + 1/6 \times 6/6 \\ 
  &= 15/36 \\
  &= 5/12
  \end{aligned}
\end{equation}

\begin{equation}
  \begin{aligned}
    P(X \leq 2) &= P(X = 1) + P(X = 2) \\
    &= 1/2 + 5/12 \\
    &= 11/12 \\
  \end{aligned}
\end{equation}

\begin{equation}
  \begin{aligned}
    P(X=3) &= P(R_1 = 1, R_2 = 1, R_3 \geq 2) \\
    &\quad + P(R_1 = 1, R_2 = 2, R_3 \geq 1) \\
    &\quad + P(R_1 = 2, R_2 = 1, R_3 \geq 1) \\
    &= P(R_1 = 1)P(R_2 = 1)P(R_3 \geq 2) \\
    &\quad + P(R_1 = 2)P(R_2 = 1)P(R_3 \geq 1) \\
    &\quad + P(R_1 = 1)P(R_2 = 2)P(R_3 \geq 1) \\
    &= 1/6 \times 1/6 \times 5/6 \\
    &\quad + 1/6 \times 1/6 \times 6/6 \\
    &\quad + 1/6 \times 1/6 \times 6/6 \\
    &= \frac{5+6+6}{216} \\
    &= 17/216
  \end{aligned}
\end{equation}

\begin{equation}
  \begin{aligned}
    P(X=4) &= P(R_1 = 1, R_2 = 1, R_3 = 1, R_4 = 1) \\
    &= (\frac{1}{6})^4 \\
    &= 1/1296
  \end{aligned}
\end{equation}

\begin{equation}
  \begin{aligned}
    E(X) &= 1P(X=1) + 2P(X=2) + 3P(X=3) + 4P(X=4) \\
    &= 1(\frac{1}{2}) + 2(\frac{5}{12}) + 3(\frac{17}{216}) + 4(\frac{1}{1296}) \\
    &= 648/1296 + 1080/1296 + 306/1296 + 4/1296 \\
    &= 2038/1296 \\
    &\approx 1.57
  \end{aligned}
\end{equation}

In [8]:
# P(X=2)
1.0/6 * 2.0/3 + 1.0/6 * 5.0/6 + 1.0/6 * 1.0

0.41666666666666663

In [9]:
# E(X)
0.5 + 2*5.0/12 + 3*17.0/216 + 4/1296.0

1.572530864197531

In [10]:
def accumulate_dots(num_dots=4, num_experiments=100000):
    def roll_dice():
        return np.random.permutation([1,2,3,4,5,6])[0]
    num_rolls_to_get_num_dots = []
    
    for i in range(num_experiments):
        _sum = 0
        count = 0
        while _sum < 4:
            _sum += roll_dice()
            count += 1
        
        num_rolls_to_get_num_dots.append(count)
        
    return pd.DataFrame({'num_rolls_to_get_num_dots': num_rolls_to_get_num_dots})

def probability(criteria, result):
    return float(criteria(result).shape[0]) / result.shape[0]
        
def expected_value(result):
    return result.mean()

In [11]:
probability(lambda x: x[x['num_rolls_to_get_num_dots'] <= 2], accumulate_dots())

0.91653

In [12]:
expected_value(accumulate_dots(num_experiments=1000000))

num_rolls_to_get_num_dots    1.587869
dtype: float64

### 2. Recall the committee example in Section 3.9.2. Suppose now, though, that the selection protocol is that there must be at least one man and at least one woman on the committee. Find $E(D)$ and $Var(D)$.

Reminder: There are 6 men and 3 women to choose from. The committee has a size of 4.

Let $M = \text{number of men in committee}$ and $W = \text{number of women in committee.}$ Let $D = M-W$, given that there must be at least one man and at least one woman on the committee.

$D$ could be $3-1=2$, $2-2=0$, or $1-3=-2$. 

Thus:

\begin{equation}
  \begin{aligned}
     E(D) &= 2P(D=2) + 0P(D=0) + -2P(D=-2) \\
          &= 2P(D=2) - 2P(D=-2) \\
          &= 2(\frac{60}{111}) - 2(\frac{6}{111}) \\
          &= 120/111 - 12/111 \\
          &= 108/111 \\
          &\approx 0.97
  \end{aligned}
\end{equation}

\begin{equation}
  \begin{aligned}
     Var(D) &= E(D^2) - E(D)^2
  \end{aligned}
\end{equation}

\begin{equation}
  \begin{aligned}
     E(D^2) &= (2)^2P(D=2) + (0)^2P(D=0) + (-2)^2P(D=-2) \\
     &= 4P(D=2) + 4P(D=-2) \\
     &= 4(\frac{60}{111}) + 4(\frac{6}{111}) \\
     &= 264/111 \\
     &\approx 2.378
  \end{aligned}
\end{equation}

Thus:

\begin{equation}
  \begin{aligned}
     Var(D) &= E(D^2) - E(D)^2 \\
     &= 264/111 - (108/111)^2 \\
     &= 156/111 \\
     &\approx 1.432
  \end{aligned}
\end{equation}

In [13]:
# P(D=2)
ms.comb(6, 3)*ms.comb(3, 1) / \
(ms.comb(6, 3)*ms.comb(3, 1) + ms.comb(6, 2)*ms.comb(3, 2) + ms.comb(6, 1)*ms.comb(3, 3))

0.54054054054054057

In [14]:
# P(D=-2)
ms.comb(6, 1)*ms.comb(3, 3) / \
(ms.comb(6, 3)*ms.comb(3, 1) + ms.comb(6, 2)*ms.comb(3, 2) + ms.comb(6, 1)*ms.comb(3, 3))

0.054054054054054057

In [15]:
# E(X)
2*(ms.comb(6, 3)*ms.comb(3, 1) / \
(ms.comb(6, 3)*ms.comb(3, 1) + ms.comb(6, 2)*ms.comb(3, 2) + ms.comb(6, 1)*ms.comb(3, 3))) \
- 2*(ms.comb(6, 1)*ms.comb(3, 3) / \
(ms.comb(6, 3)*ms.comb(3, 1) + ms.comb(6, 2)*ms.comb(3, 2) + ms.comb(6, 1)*ms.comb(3, 3)))

0.97297297297297303

In [16]:
def generate_committee(num_experiments=10000):
    # Note: not the same as choosing 1 man, 
    # choosing 1 woman, then choosing 2 people from the remaining people.
  
    men_count_minus_women_count = []
    people_to_choose_from = ['M','M','M','M','M','M', 'W','W','W']
    
    def count_men_minus_women(people_arr):
        men_count = 0
        women_count = 0
        
        for i in people_arr:
            if i == 'M':
                men_count += 1
            else:
                women_count += 1
        
        return men_count - women_count
    
    experiment_count = 0
    while experiment_count < num_experiments:
        permute = np.random.permutation(people_to_choose_from)[0:4]
        men_minus_women = count_men_minus_women(permute)
        
        if men_minus_women == -2 or men_minus_women == 0 or men_minus_women == 2:
            experiment_count +=1
            men_count_minus_women_count.append(men_minus_women)
            
    return pd.DataFrame({'men_count_minus_women_count': men_count_minus_women_count})    

In [17]:
def variance(df, column_name):
    return ((df[column_name] - df[column_name].mean())**2).sum() / df.shape[0]

In [18]:
expected_value(generate_committee())

men_count_minus_women_count    0.964
dtype: float64

In [19]:
# Var(X)
264.0/111 - (108.0/111)**2

1.4317019722425128

In [20]:
variance(generate_committee(), 'men_count_minus_women_count')

1.4116158400000001

In [21]:
gen_committee = generate_committee(num_experiments=10000)
prob_x_is_2 =     probability(lambda x: x[x['men_count_minus_women_count'] == 2], gen_committee)
prob_x_is_neg_2 = probability(lambda x: x[x['men_count_minus_women_count'] == -2], gen_committee)
print "P(X=2) = {}, P(X=-2) = {}".format(prob_x_is_2, prob_x_is_neg_2)

P(X=2) = 0.5464, P(X=-2) = 0.0516


In [22]:
4*60/111.0 + 4*6/111.0

2.3783783783783785

In [23]:
ms.comb(4,2)

6.0

### 3. Suppose a bit stream is subject to errors, with each bit having probability p of error, and with the bits being independent. Consider a set of four particular bits. Let $X$ denote the number of erroneous bits among those four.

#### (a) Find $P(X = 2)$ and $EX$.

\begin{equation}
  \begin{aligned}
     P(X=2) = {4 \choose 2}p^2(1-p)^2
  \end{aligned}
\end{equation}

We could represent X as a sum of indicator variables $G_i$: $X = G_1 + G_2 + G_3 + G_4$:

\begin{equation}
  \begin{aligned}
          X  &= G_1 + G_2 + G_3 + G_4 \\
        E(X) &= E(G_1 + G_2 + G_3 + G_4) \\
             &= E(G_1) + E(G_2) + E(G_3) + E(G_4) \\
             &= p + p + p + p \\
             &= 4p \\
  \end{aligned}
\end{equation}


#### (b) What famous parametric family of distributions does the distribution of X belong to? 
This distribution belongs to Binomial family of distributions.

#### (c) Let $Y$ denote the maximum number of consecutive erroneous bits. Find $P(Y = 2)$ and $Var(Y)$.

There are five possible configurations where $Y=2$. They are as follows: $1100, 0110, 0011, 1101, 1011$, where $1$ stood for an error and $0$ for none. Thus:

\begin{equation}
  \begin{aligned}
     P(Y=2) &= 3p^2(1-p)^2 + 2p^3(1-p)
  \end{aligned}
\end{equation}

$Var(Y) = E(Y^2)-E(Y)^2$, so we need to calculate $E(Y^2)$ and $E(Y)^2$. To do so, we need to calculate $P(Y=k)$.

There are seven possible configurations where $Y=1$. They are as follows: $1000, 0100, 0010, 0001, 1010, 0101, 1001$, where $1$ stood for an error and $0$ for none. Thus:

\begin{equation}
  \begin{aligned}
     P(Y=1) = 4p(1-p)^3 + 3p^2(1-p)^2
  \end{aligned}
\end{equation}

There are two possible configurations where $Y=3$. They are as follows: $1110, 0111$, where $1$ stood for an error and $0$ for none. Thus:

\begin{equation}
  \begin{aligned}
     P(Y=3) = 2p^3(1-p)
  \end{aligned}
\end{equation}

There is only one possible configuration where $Y=4$. They are as follows: $1111$ where $1$ stood for an error and $0$ for none. Thus:

\begin{equation}
  \begin{aligned}
     P(Y=4) = p^4
  \end{aligned}
\end{equation}

\begin{equation}
  \begin{aligned}
     E(Y) &= \sum_{k}kP(Y=k) \\
     &= 1(4p(1-p)^3 + 3p^2(1-p)^2) + 2(3p^2(1-p)^2 + 2p^3(1-p)) + 3(2p^3(1-p)) + 4(p^4) \\
     &= 4p(1-p)^3 + 3p^2(1-p)^2 + 6p^2(1-p)^2 + 4p^3(1-p) + 6p^3(1-p) + 4(p)^4 \\
     &= 4p(1-p)^3 + 9p^2(1-p)^2 + 4p^3(1-p) + 6p^3(1-p) + 4(p)^4 \\
  \end{aligned}
\end{equation}

\begin{equation}
  \begin{aligned}
     E(Y)^2 &= (4p(1-p)^3 + 9p^2(1-p)^2 + 4p^3(1-p) + 6p^3(1-p) + 4(p)^4)^2 \\
  \end{aligned}
\end{equation}

\begin{equation}
  \begin{aligned}
     E(Y^2) &= \sum_kk^2P(Y=k) \\
     &= 1^2(4p(1-p)^3 + 3p^2(1-p)^2) + 2^2(3p^2(1-p)^2 + 2p^3(1-p)) + 3^2(2p^3(1-p)) + 4^2(p^4) \\
     &= (4p(1-p)^3 + 3p^2(1-p)^2) + 4(3p^2(1-p)^2 + 2p^3(1-p)) + 9(2p^3(1-p)) + 16(p^4)  \\
     &= 4p(1-p)^3 + 3p^2(1-p)^2 + 12p^2(1-p)^2 + 8p^3(1-p) + 18p^3(1-p) + 16(p^4)  \\
     &= 4p(1-p)^3 + 15p^2(1-p)^2 + 26p^3(1-p) + 16(p^4)  \\
  \end{aligned}
\end{equation}

\begin{equation}
  \begin{aligned}
     Var(Y) &= 4p(1-p)^3 + 15p^2(1-p)^2 + 26p^3(1-p) + 16(p^4) - (4p(1-p)^3 + 9p^2(1-p)^2 + 4p^3(1-p) + 6p^3(1-p) + 4(p)^4)^2 \\
  \end{aligned}
\end{equation}

In [24]:
def consecutive_bits(num_experiments=10000, prob_failure=0.7):
    num_bits = 4
    num_consecutive_bits = []
    
    def generate_bits():
        return np.random.binomial(1, prob_failure, num_bits)
    
    def count_max_consecutive(bits):
        
        consecutive = False
        max_count = 0
        for bit in bits:
            if bit == 1:
                if consecutive==False:
                    if max_count == 0:
                        max_count += 1
                    consecutive = True
                else:
                    max_count += 1
                    max_count
            else:
                consecutive = False
#         if max_count == 2:
#             print "bits: {}, max_count: {}".format(bits, max_count)
        return max_count
    
    for i in range(num_experiments):
        num_consecutive_bits.append(count_max_consecutive(generate_bits()))
        
    return pd.DataFrame({'num_consecutive_bits': num_consecutive_bits})
        
        
    

In [25]:
# P(Y=2)
3*0.5**4

0.1875

In [26]:
consecutive_bits(num_experiments=100, prob_failure=0.5)

num_consecutive_bits
0                      0
1                      0
2                      2
3                      1
4                      0
5                      1
6                      1
7                      1
8                      1
9                      2
10                     2
11                     2
12                     3
13                     3
14                     1
15                     2
16                     1
17                     1
18                     2
19                     2
20                     2
21                     1
22                     2
23                     1
24                     1
25                     1
26                     1
27                     1
28                     1
29                     2
..                   ...
70                     2
71                     1
72                     1
73                     1
74                     4
75                     2
76                     1
77                     1
78                     2
79                     1
80                     2
81                     0
82                     3
83                     2
84                     1
85                     3
86                     1
87                     1
88                     3
89                     3
90                     1
91                     2
92                     1
93                     0
94                     2
95                     1
96                     1
97                     3
98                     2
99                     1

[100 rows x 1 columns]

In [27]:
expected_value(consecutive_bits(num_experiments=100, prob_failure=0.5))

num_consecutive_bits    1.6
dtype: float64

In [28]:
# P(Y=1)
4*0.5*0.5**3 + 3*0.5**2*0.5**2

0.4375

In [29]:
# P(Y=1)
probability(lambda x: x[x.num_consecutive_bits == 1], consecutive_bits(num_experiments=100000, prob_failure=0.5))

0.43659

In [30]:
# P(Y=2)
5*0.5**2*0.5**2

0.3125

In [31]:
# P(Y=2)
probability(lambda x: x[x.num_consecutive_bits == 2], consecutive_bits(num_experiments=1000000, prob_failure=0.5))

0.312388

In [32]:
# P(Y=3) = 2*0.5^3*(1-0.5)
2*0.5**3*0.5

0.125

In [33]:
# P(Y=3)
probability(lambda x: x[x.num_consecutive_bits == 3], consecutive_bits(num_experiments=100000, prob_failure=0.5))

0.12378

In [34]:
# P(Y=4) = 0.5**4
0.5**4

0.0625

In [35]:
# P(Y=4)
probability(lambda x: x[x.num_consecutive_bits == 4], consecutive_bits(num_experiments=100000, prob_failure=0.5))

0.06229

In [36]:
# Var(Y)
variance(consecutive_bits(num_experiments=100000, prob_failure=0.5), 'num_consecutive_bits')

0.96115318039999986

In [37]:
def variance_consecutive_bits(prob_failure=0.5):
    p = prob_failure
    return 4*p*(1-p)**3 + 15*p**2*(1-p)**2 + 26*p**3*(1-p) + 16*p**4 \
        - (4*p*(1-p)**3 + 9*p**2*(1-p)**2 + 4*p**3*(1-p) + 6*p**3*(1-p) + 4*p**4)**2
    

In [38]:
# Var(Y)
variance_consecutive_bits(prob_failure=0.5)

0.96484375

# 

### 4. Derive $Var(W) = \frac{1-p}{p^2}$.

\begin{equation}
  \begin{aligned}
     E(W) = 1/p
  \end{aligned}
\end{equation}

\begin{equation}
  \begin{aligned}
     E(W^2) = \sum_{i=1}^{\infty}i^2(1-p)^{(i-1)}p
  \end{aligned}
\end{equation}

Let $1-p=q$:

\begin{equation}
  \begin{aligned}
     E(W^2) &= \sum_{i=1}^{\infty}i^2(q)^{(i-1)}p + \sum_{i=1}^{\infty}i(q)^{(i-1)}p - \sum_{i=1}^{\infty}i(q)^{(i-1)}p \\
     &= \sum_{i=1}^{\infty}i(i+1)(q)^{(i-1)}p - \sum_{i=1}^{\infty}i(q)^{(i-1)}p \\
     &= p\sum_{i=1}^{\infty}i(i+1)(q)^{(i-1)} - p\sum_{i=1}^{\infty}i(q)^{(i-1)} \\
  \end{aligned}
\end{equation}

Let's solve for $p\sum_{i=1}^{\infty}i(q)^{(i-1)}$:

\begin{equation}
  \begin{aligned}
     \sum_{i=1}^{\infty}q^ip &= p\sum_{i=1}^{\infty}q^i \\
     p\sum_{i=1}^{\infty}q^i) &= \frac{p}{1-q} \\
     \frac{d}{dq}(p\sum_{i=1}^{\infty}q^i) &= \frac{d}{dq}(\frac{p}{1-q}) \\
     (p\sum_{i=1}^{\infty}iq^{i-1})   &= \frac{p}{(1-q)^2} \\
     &= \frac{p}{(1-(1-p))^2} \\
     &= p/p^2 \\
     &= 1/p
  \end{aligned}
\end{equation}

So far, we've simplified a piece of $E(W^2)$:

\begin{equation}
  \begin{aligned}
     E(W^2) &= p\sum_{i=1}^{\infty}i(i+1)(q)^{(i-1)} - p\sum_{i=1}^{\infty}i(q)^{(i-1)} \\
            &=p\sum_{i=1}^{\infty}i(i+1)(q)^{(i-1)} - 1/p \\
  \end{aligned}
\end{equation}

Let's then solve for $p\sum_{i=1}^{\infty}i(i+1)(q)^{(i-1)}$:

\begin{equation}
  \begin{aligned}
     p\sum_{j=1}^{\infty}q^j &= \frac{1}{1-q} \\
          \frac{d}{dq}(p\sum_{j=1}^{\infty}q^j) &= \frac{d}{dq}(\frac{1}{1-q}) \\
          p\sum_{j=1}^{\infty}jq^{j-1}&= \frac{1}{(1-q)^2} \\
          \frac{d}{dq}(p\sum_{j=1}^{\infty}jq^{j-1}) &= \frac{d}{dq}(\frac{1}{(1-q)^2}) \\
           p\sum_{j=1}^{\infty}j(j-1)q^{j-2} &= \frac{2}{(1-q)^3}
  \end{aligned}
\end{equation}

Let $j=i-1$:

\begin{equation}
  \begin{aligned}
     p\sum_{i=1}^{\infty}i(i+1)(q)^{(i-1)} &= p\sum_{j=2}^{\infty}(j-1)(j)(q)^{(j-2)} \\
            &= p\frac{2}{(1-q)^3} \\
            &= p\frac{2}{(1-(1-p))^3} \\
            &= \frac{2p}{p^3}\\
            &= \frac{2}{p^2} \\
  \end{aligned}
\end{equation}

Thus:

\begin{equation}
  \begin{aligned}
     E(W^2) &= p\sum_{i=1}^{\infty}i(i+1)(q)^{(i-1)} - p\sum_{i=1}^{\infty}i(q)^{(i-1)} \\
            &= \frac{2}{p^2} - 1/p \\
            &= \frac{2-p}{p^2} \\
  \end{aligned}
\end{equation}

Therefore, $Var(W)$ is as follows:

\begin{equation}
  \begin{aligned}
     Var(W) &= E(W^2) - E(W)^2 \\
            &= \frac{2-p}{p^2} - (\frac{1}{p})^2 \\
            &= \frac{1-p}{p^2} \\
  \end{aligned}
\end{equation}



### 5. Finish the computation in 4.18

 
\begin{equation}
  \begin{aligned}
    ED &= \sum_{i=1}^{10}(11-i)(1-0.15)^{i-1}0.15 + \sum_{i=11}^{\infty}(i-11)(1-0.15)^{i-1}0.15 \\
    &= \sum_{i=1}^{10}(11-i)(1-0.15)^{i-1}0.15 + \sum_{i=1}^{\infty}(i-11)(1-0.15)^{i-1}0.15 \\
    &\quad - \sum_{i=1}^{10}(i-11)(1-0.15)^{i-1}0.15 \\
    &= \sum_{i=1}^{10}(11-i)(1-0.15)^{i-1}0.15 + \sum_{i=1}^{\infty}(i-11)(1-0.15)^{i-1}0.15 \\
    &\quad + \sum_{i=1}^{10}(11-i)(1-0.15)^{i-1}0.15 \\
    &= 2\sum_{i=1}^{10}(11-i)(1-0.15)^{i-1}0.15 + \sum_{i=1}^{\infty}(i-11)(1-0.15)^{i-1}0.15 \\
    &= 0.3\sum_{i=1}^{10}(11-i)(1-0.15)^{i-1} + 0.15\sum_{i=1}^{\infty}(i-11)(1-0.15)^{i-1} \\
    &= 0.3\sum_{i=1}^{10}(11-i)(1-0.15)^{i-1} + 0.15(\sum_{i=1}^{\infty}(i)(1-0.15)^{i-1} - 11\sum_{i=1}^{\infty}(1-0.15)^{i-1}) \\
    &= 0.3\sum_{i=1}^{10}(11-i)(1-0.15)^{i-1} + 0.15(\frac{1}{(1-(1-0.15))^2} - 11\sum_{j=0}^{\infty}(1-0.15)^{j}) \\
    &= 0.3\sum_{i=1}^{10}(11-i)(1-0.15)^{i-1} + 0.15(\frac{1}{0.15^2} - \frac{11}{0.15}) \\
    &= 0.3\sum_{i=1}^{10}(11-i)(1-0.15)^{i-1} + \frac{1}{0.15} - 11  \\
    &= 0.3\sum_{i=1}^{10}(11-i)(1-0.15)^{i-1} + -4.\overline{3} \\
    &\approx 10.897909915861522 - 4.\overline{3} \\
    &\approx 6.56
  \end{aligned}
\end{equation}




In [39]:
def sum_from_1_to_10_of_11_minus_i_times_1_minus_p_raised_to_i_minus_1_times_p(p=0.15, end=10):
    _sum = 0
    for i in range(1,end+1):
        _sum += (11 - i)*(1-p)**(i-1)*p
    return _sum
2 * sum_from_1_to_10_of_11_minus_i_times_1_minus_p_raised_to_i_minus_1_times_p(p=0.15, end=10)

10.897909915861522

In [71]:
def park_sim(num_experiments=100000, p=0.15):
    df = pd.DataFrame({'num_parking_spaces'\
                       : np.random.geometric(p=p, 
                                             size=num_experiments)})
    df['num_open_spaces_first_block'] = np.random.binomial(n=10,
                                                           p=p,
                                                           size=num_experiments)
    df['dist_from_dest'] = df.num_parking_spaces.apply(lambda x: abs(11-x))
    return df

park_sim(num_experiments=1000000).mean()

num_parking_spaces             6.654932
num_open_spaces_first_block    1.501024
dist_from_dest                 6.557950
dtype: float64

In [41]:
# P(D=1)
(1-0.15)**(10-1)*.15 + (1-0.15)**(12-1)*0.15

0.05984402849592259

In [42]:
probability(lambda x: x[x['dist_from_dest'] == 1], park_sim())

0.05957

### 6. Derive the facts that for a Poisson-distributed random variable $X$ with parameter $λ$, $EX = Var(X) = λ$. Use the hints in Section 4.5.

$P(X=k) = \frac{e^{−λ}λ^k}{k!}, k = 0,1,2,...$

First let's show that $EX = \lambda$:

\begin{equation}
  \begin{aligned}
    e^λ &= \sum_{k=0}^{\infty}\frac{λ^k}{k!} \\
    \frac{d}{dλ}(e^λ) &= \frac{d}{dλ}(\sum_{k=0}^{\infty}\frac{λ^k}{k!}) \\
    e^λ &= \sum_{k=0}^{\infty}k\frac{λ^{k-1}}{k!} \\
    λe^{-λ}(e^λ) &= λe^{-λ}(\sum_{k=0}^{\infty}k\frac{λ^{k-1}}{k!}) \\
    λ &= \sum_{k=0}^{\infty}k\frac{λ^{k}}{k!}e^{-λ} \\
    &= \sum_{k=0}^{\infty}kP(X=k)\\
    &= EX
  \end{aligned}
\end{equation}

Then let's show that $Var(X) = \lambda$:

\begin{equation}
  \begin{aligned}
    Var(X) &= E(X^2) - E(X)^2 \\ 
    &= E(X^2) - \lambda^2 \\
  \end{aligned}
\end{equation}

Let's evaluate $E(X^2)$:

\begin{equation}
  \begin{aligned}
        E(X^2) &= \sum_{k=0}^{\infty}k^2P(X=k)  \\
    &= \sum_{k=0}^{\infty}k^2\frac{e^{−λ}λ^k}{k!} \\
    &= \sum_{k=0}^{\infty}k\frac{e^{−λ}λ^k}{(k-1)!} \\
    &= \sum_{k=1}^{\infty}k\frac{e^{−λ}λ^k}{(k-1)!} && \text{When $k=0$, term is 0}\\
    &= \lambda e^{−λ}\sum_{k=1}^{\infty}k\frac{λ^{k-1}}{(k-1)!} && \text{Factor out} \\
    &= \lambda e^{−λ}\sum_{k=1}^{\infty}(k-1)\frac{λ^{k-1}}{(k-1)!} + \lambda e^{-\lambda}\sum_{k=1}^{\infty}\frac{λ^{k-1}}{(k-1)!} \\
    &= \lambda e^{−λ}\sum_{k=2}^{\infty}(k-1)\frac{λ^{k-1}}{(k-1)!} + \lambda e^{-\lambda}\sum_{k=1}^{\infty}\frac{λ^{k-1}}{(k-1)!} && \text{When $k=1$ for the first summation, it becomes $0$}\\
    &= \lambda e^{−λ}\sum_{k=2}^{\infty}\frac{λ^{k-1}}{(k-2)!} + \lambda e^{-\lambda}\sum_{k=1}^{\infty}\frac{λ^{k-1}}{(k-1)!} && \text{Simplify the first term summation}\\
    &= \lambda^2 e^{−λ}\sum_{k=2}^{\infty}\frac{λ^{k-2}}{(k-2)!} + \lambda e^{-\lambda}\sum_{k=1}^{\infty}\frac{λ^{k-1}}{(k-1)!} && \text{Extract $\lambda$}\\    
    &= \lambda^2 e^{−λ}\sum_{m=0}^{\infty}\frac{λ^{m}}{m!} + \lambda e^{-\lambda}\sum_{n=0}^{\infty}\frac{λ^{n}}{n!} && \text{Let $m=k-2$ and $n=k-1$}\\
    &= \lambda^2 e^{−λ}e^\lambda + \lambda e^{-\lambda}e^\lambda && \text{Substitute}\\
    &= \lambda^2 + \lambda
  \end{aligned}
\end{equation}

Going back to $Var(X)$:

\begin{equation}
  \begin{aligned}
    Var(X) &= E(X^2) - E(X)^2 \\ 
    &= E(X^2) - \lambda^2 \\
    &= \lambda^2 + \lambda - \lambda^2 \\
    &= \lambda
  \end{aligned}
\end{equation}



## 7. A civil engineer is collecting data on a certain road. She needs to have data on $25$ trucks, and $10$ percent of the vehicles on that road are trucks. State the famous parametric family that is relevant here, and find the probability that she will need to wait for more than $200$ vehicles to pass before she gets the needed data.

Essentially, we want to get 24 trucks from the first 199 cars passing by. The 200th vehicle should be the 25th truck.

We could use the negative binomial distribution here. Let $X$ be the amount of vehicles needed to get the 25th truck:

\begin{equation}
\begin{aligned}
    P(X=200) &= {n - 1 \choose k - 1}p^{k-1}(1-p)^{(n-1)-(k-1)}p \\
    &= {200 - 1 \choose 25 - 1}p^{25 - 1}(1-p)^{199-24}p \\
        &= {200 - 1 \choose 25 - 1}p^{25 - 1}(1-p)^{199-24}p \\
        &\approx 0.0056
\end{aligned}
\end{equation}

In [43]:
# P(X=200)
ms.comb(199,24)*0.1**24*(1-0.1)**(199-24)*0.1

0.0055545921220223208

In [44]:
# P(X=200)
prob_truck = 0.1
num_trucks = 25
nbinom = ss.nbinom(n=num_trucks, p=prob_truck)
num_failures = 175
nbinom.pmf(num_failures)

0.0055545921220225654

## 8. In the ALOHA example:

### (a) Find $E(X_1)$ and $Var(X_1)$, for the case $p = 0.4$, $q = 0.8$. You are welcome to use quantities already computed in the text, e.g. $P(X_1 = 1) = 0.48$, but be sure to cite equation numbers

\begin{equation}
\begin{aligned}
E(X_1) &= 0P(X=0) + 1P(X=1) + 2P(X=2) \\
&= 1 \times 0.48 + 2 \times 0.52 && \text{See 2.1 for $P(X=2)$}\\ 
&= 1.52
\end{aligned}
\end{equation}

\begin{equation}
\begin{aligned}
Var(X_1) &= E(X_1^2) - E(X_1)^2 \\
&= 0^2P(X=0) + 1^2P(X=1) + 2^2P(X=2) - 1.52^2\\
&= 1 \times 0.48 + 4 \times 0.52 - 2.3104\\ 
&= 0.2496
\end{aligned}
\end{equation}



In [45]:
# E(X)
1*0.48 + 2*0.52

1.52

In [46]:
# E(X)^2
1.52**2

2.3104

In [47]:
# Var(X)
0.48 + 4*0.52 - 2.3104

0.24960000000000004

###(b) Find $P(\text{collision during epoch 1})$ for general $p$, $q$.

Given that in the beginning both nodes were active, collision during epoch 1 could only happen when both active nodes were to send at the same time, e.g. $p^2$.

## 9. Our experiment is to toss a nickel until we get a head, taking $X$ rolls, and then toss a dime until we get a head, taking $Y$ tosses. Find:

### (a) $Var(X+Y)$

\begin{equation}
\begin{aligned}
Var(X + Y) &= Var(X) + Var(Y) + 2Cov(X,Y) && \text{Definition} \\
&= Var(X) + Var(Y) && \text{$2Cov(X,Y) = 0$ because X and Y are independent}\\
&= \frac{1-p}{p^2} + \frac{1-q}{q^2} && \text{Both are geometric distributions}\\ 
&= \frac{0.5}{0.25} + \frac{0.5}{0.25} && \text{Both coins are assumed to be fair coins}\\
&= 2 + 2 \\
&= 4
\end{aligned}
\end{equation}


In [48]:
def toss_nickel_and_dime(num_experiments=10000):
    X = np.random.geometric(p=0.5, size=num_experiments)
    Y = np.random.geometric(p=0.5, size=num_experiments)
    
    return pd.DataFrame({'X+Y': X+Y, 'X': X, 'X**2': X**2})  

variance(toss_nickel_and_dime(num_experiments=1000000), 'X+Y')

3.9994258066309989

### (b) Long-run average in a “notebook” column labeled $X^2$.

$E(X^2) \approx 5.99$

In [49]:
expected_value(toss_nickel_and_dime(num_experiments=1000000))

X       1.998646
X**2    5.980706
X+Y     3.999462
dtype: float64

## 10. Consider the game in Section 4.8. Find $E(Z)$ and $Var(Z)$, where $Z = Y_6 − X_6$.

\begin{equation}
\begin{aligned}
Z &= Y_6 - X_6 \\
E(Z) &= E(Y_6 - X_6) \\
&= E(Y_6) - E(X_6) \\
&= (4p) - (6p) \\
&= -2p \\
&= -2(0.5) \\
&= -1
\end{aligned}
\end{equation}

Note that $Y_6$ stands for the number of heads Jill gets in 6 rounds (4 tosses). $X_6$ is the number of heads Jack gets in 6 rounds (6 tosses).

In [50]:
def jack_and_jill_toss(num_experiments=10000):
    X = np.random.binomial(n=6, p=0.5, size=num_experiments)
    Y = np.random.binomial(n=4, p=0.5, size=num_experiments)
    
    return pd.DataFrame({'Z': Y-X})  

expected_value(jack_and_jill_toss(num_experiments=1000000))

Z   -1.000332
dtype: float64

\begin{equation}
\begin{aligned}
Z &= Y_6 - X_6 \\
Var(Z) &= Var(Y_6 - X_6) \\
&= Var(Y_6) + Var(-X_6) + 2Cov(Y_6, -X_6) \\
&= Var(Y_6) + Var(-X_6) && \text{Jack's tosses are independent from Jill's} \\
&= 4p(1-p) + (-1)^2 6p(1-p) && \text{See Equation 3.49} \\
&= 10p(1-p) \\
&= 10(0.5)(1-(0.5)) && \text{Assuming fair coins} \\
&= 2.5
\end{aligned}
\end{equation}

In [51]:
np.insert(arr=np.zeros(6),obj=5,values=[1,2])

array([ 0.,  0.,  0.,  0.,  0.,  1.,  2.,  0.])

In [52]:
variance(jack_and_jill_toss(), 'Z')

2.50684079

## 11. Say we choose six cards from a standard deck, one at a time WITHOUT replacement. Let $N$ be the number of kings we get. Does $N$ have a binomial distribution? Choose one: (i) Yes. (ii) No, since trials are not independent. (iii) No, since the probability of success is not constant from trial to trial. (iv) No, since the number of trials is not fixed. (v) (ii) and (iii). (iv) (ii) and (iv). (vii) (iii) and (iv).

We know that trials are not independent. For example, if we happen to pick a king for the first card, then we know that there is one less king in the deck, and thus the probability of getting a king in the second pick is totally dependent on the first, so that eliminates the first option.

The probability of success is not constant from trial to trial. This is just another way of rephasing the previous statement. Again, if you pick a king on the first trial, we know that the probability of getting a king later on is smaller.

Number of trials is fixed. It is always 6, so the fourth option is not true.

Thus the answer should be the fifth option: (ii) and (iii).

## 12. Suppose we have $n$ independent trials, with the probability of success on the $i^{th}$ trial being $p_i$. Let $X = \text{the number of successes}$. Use the fact that “the variance of the sum is the sum of the variance” for independent random variables to derive $Var(X)$.

\begin{equation}
\begin{aligned}
p_0 &= p_1 = ... = p_n && \text{because each trial is independent of the others}\\
X &= B_1 + B_2 + ... + B_n && \text{$B_i$ is the i$^{th}$ indicator variable, with probability $p$} \\
Var(X) &= Var(B_1 + B_2 + ... + B_n) \\
&= Var(B_1) + Var(B_2) + ... + Var(B_n) \\ 
&= nVar(B) && \text{Since each indicator variable has the same probability $p$ of being true.}\\ 
&= np(1-p) && \text{See right below for details} \\
\end{aligned}
\end{equation}

\begin{equation}
\begin{aligned}
Var(B) &= E(B^2) - E(B)^2 \\
&= (0^2P(B=0) + 1^2P(B=1)) - E(B)^2 \\
&= P(B=1) - E(B)^2 \\
&= p - E(B)^2 \\
&= p - (0P(B=0) + 1P(B=1))^2 \\
&= p - p^2 \\
&= p(1-p) \\
\end{aligned}
\end{equation}

## 13. Prove Equation 3.41

\begin{equation}
\begin{aligned}
Var(U) &= E((U-EU)^2) \\
&= E(U^2 - UEU - UEU + (EU)^2) \\
&= E(U^2 - 2UEU + (EU)^2) \\
&= E(U^2 - 2UEU + EUEU) \\
&= E(U^2) - E(2UEU) + E(EUEU) \\
&= E(U^2) - 2EU(E(U)) + E(EUEU) && \text{factor out constants}\\
&= E(U^2) - 2EUEU + EUEU && \text{expected value of a constant is itself}\\
&= E(U^2) - 2E(U)^2 + E(U)^2 \\ 
&= E(U^2) - E(U)^2 \\
\end{aligned}
\end{equation}

## 14. Show that if X is a nonnegative-integer valued random variable, then

\begin{equation}
\begin{aligned}
    EX = \sum_{i=1}^{\infty}P(X \geq i)
\end{aligned}
\end{equation}

Hint: Write $i = \sum_{j=1}^i1$, and when you see an iterated sum, reverse the order of summation.

\begin{equation}
\begin{aligned}
    \sum_{i=1}^{\infty}P(X \geq i) &= P(X\geq1) + P(X\geq2) + P(X\geq3) + ... \\
    &= P(X=1)+P(X=2)+P(X=3)+ ... && P(X\geq1) \\
    &\quad\quad\quad\quad + P(X=2)+P(X=3)+ ... && P(X\geq2)  \\
    &\quad\quad\quad\quad\quad\quad\quad\quad + P(X=3)+ ... && P(X\geq3) \\
    &= 1P(X=1) + 2P(X=2) + 3P(X=3) + ... \\
    &= \sum_{i=1}^{\infty}iP(X=i)\\
    &= EX \\
\end{aligned}
\end{equation}

In [53]:
ms.comb(1000,500)*0.5**1000

0.02522501817837652

## 15. Suppose we toss a fair coin n times, resulting in X heads. Show that the term expected value is a misnomer, by showing that

\begin{equation}
\begin{aligned}
    \lim_{n \to \infty}P(X=n/2)=0
\end{aligned}
\end{equation}

## Use Stirling's approximation,

\begin{equation}
\begin{aligned}
    k! = \sqrt{2\pi k}\left(\frac{k}{e}\right)^k
\end{aligned}
\end{equation}

Expected Value:

\begin{equation}
\begin{aligned}
    E(X) &= \sum_{i=1}^{\infty}iP(X=i) \\
    &= np && \text{Expected value of binomial distribution}\\
    &= n/2
\end{aligned}
\end{equation}

Probability Mass Function of Binomial Distribution:

\begin{equation}
\begin{aligned}
    P(X=k) &= {n \choose k}p^k(1-p)^{n-k} \\
    P(X=n/2) &= {n \choose n/2}p^{n/2}(1-p)^{n-n/2} && \text{Substitute $n/2$ for $k$}\\
    &= {n \choose n/2}0.5^{n/2}0.5^{n/2} && \text{Fair coin assumption}\\
    &= {n \choose n/2}0.5^{n} \\
    &= \frac{n!}{(n-n/2)!(n/2)!}0.5^n \\
        &= \frac{n!}{((n/2)!)^2}0.5^n \\
    &= \frac{\sqrt{2\pi n}\left(\frac{n}{e}\right)^n}{(\sqrt{2\pi n/2}\left(\frac{n/2}{e}\right)^{n/2})^2}0.5^{n} && \text{Use Stirling's approximation} \\
    &= \frac{\sqrt{2\pi n}\left(\frac{n}{e}\right)^n}{\pi n\left(\frac{n/2}{e}\right)^{n}}0.5^{n} \\
    &= \frac{\sqrt{2\pi n}\left(\frac{n}{e}\right)^n}{\pi n\left(\frac{n/2}{e}\right)^{n}}0.5^{n} \\
    &= \frac{\sqrt{2\pi n}\left(\frac{n/2}{e}\right)^n}{\pi n\left(\frac{n/2}{e}\right)^{n}} \\
    &= \frac{\sqrt{2\pi n}}{\pi n} \\
    \lim_{n \to \infty}P(X=n/2) &= \lim_{n \to \infty} \frac{\sqrt{2\pi n}}{\pi n} \\
    &= 0
\end{aligned}
\end{equation}

## 16. Suppose $X$ and $Y$ are independent random variables with standard deviations $3$ and $4$, respectively.

### (a) Find $Var(X+Y)$.

\begin{equation}
\begin{aligned}
    Var(X+Y) &= Var(X) + Var(Y) + 2Cov(X,Y) \\
    &= Var(X) + Var(Y) && \text{X and Y are independent} \\
    &= 3 + 4 \\
    &= 7
\end{aligned}
\end{equation}

### (b) Find $Var(2X - Y)$.

\begin{equation}
\begin{aligned}
    Var(2X-Y) &= (2)^2Var(X) + (-1)^2Var(Y) + 2(2)(-1)Cov(X,Y) \\
    &= 4Var(X) + Var(Y) && \text{X and Y are independent} \\
    &= 4(3)+ 4 \\
    &= 16
\end{aligned}
\end{equation}

## 18. Let $X$ be the total number of dots we get if we roll three dice. Find an upper bound for $P(X \geq 15)$, using our course materials.

\begin{equation}
\begin{aligned}
    P(X \geq d) &\leq \frac{EX}{d} && \text{Markov's Inequality} \\
    P(X \geq 15) &\leq \frac{EX}{15} && \text{d=15} \\
    & \leq \frac{E(A+B+C)}{15} && \text{$A$, $B$, $C$ are the number of dots on each specific die} \\
    & \leq \frac{E(A) + E(B) + E(C)}{15} \\
    & \leq \frac{3.5 + 3.5 + 3.5}{15} && \text{Each roll has an expected value of 3.5}\\
    & \leq \frac{10.5}{15} && \text{Each roll has an expected value of 3.5}\\
    & \leq 0.7 \\
\end{aligned}
\end{equation}


In [57]:
def sum_three_dice_rolls(num_experiments=10000):
    def roll_die():
        vals = [1,2,3,4,5,6]
        return np.random.permutation(vals)[0]
    
    results = []
    for i in range(num_experiments):
        results.append(roll_die() + roll_die() + roll_die())
        
    return pd.DataFrame({'sum_three_dice_rolls': results})
        

In [59]:
# should be less than or equal to 0.7
probability(lambda x: x[x['sum_three_dice_rolls'] >= 15], sum_three_dice_rolls())

0.0891

## 19. Suppose $X$ and $Y$ are independent random variables, and let $Z = XY$. Show that $Var(Z) = E(X^2)E(Y^2) − [E(X)]^2[E(Y )]^2$.

\begin{equation}
\begin{aligned}
    E(AB) &= \sum_j \sum_k jkP(A=j, B=k) \\
    &= \sum_j \sum_k jkP(A=j)P(B=k) && \text{by independence assumption} \\
    &= \sum_j jP(A=j) \sum_k kP(B=k) \\
    &= E(A)E(B) && \text{Expected value of two independent random variables}
\end{aligned}
\end{equation}

\begin{equation}
\begin{aligned}
    Var(Z) &= E(Z^2)-E(Z)^2 \\
    Var(XY) &= E((XY)^2) - E(XY)^2 \\
    &= E(X^2Y^2) - E(XY)E(XY) \\
    &= E(X^2)E(Y^2) - E(XY)E(XY) && \text{Use expected value of two independent random variables}\\
    &= E(X^2)E(Y^2) - E(X)E(Y)E(X)E(Y) && \text{Again, use expected value of two independent random variables}\\
    &= E(X^2)E(Y^2) - E(X)^2E(Y)^2
\end{aligned}
\end{equation}

## 20. This problem involves a very simple model of the Web. (Far more complex ones exist.) 
## Suppose we have $n$ Websites. For each pair of sites $i$ and $j$, $i \neq j$, there is a link from site $i$ to site $j$ with probability $p$, and no link (in that direction) with probability $1-p$. Let $N_i$ denote the number of sites that site $i$ is linked to; note that $N_i$ can range from $0$ to $n-1$. Also, let $M_{ij}$ denote the number of outgoing links that sites $i$ and $j$ have in common, not counting the one between them, if any. Assume that each site forms its outgoing links independently of the others.

Say n = 10, p = 0.2. Find the following:

### (a) $P(N_1 =3)$

\begin{equation}
\begin{aligned}
    P(N_1 = k) &= {n \choose k}p^k(1-p)^{n-k} \\
    P(N_1 = 3) &= {10 \choose 3}0.2^3(1-0.2)^{10-3} \\
    &\approx 0.2013
\end{aligned}
\end{equation}

In [63]:
def prob_sites_links_to(num_sites=3):
    binom = ss.binom(n=10, p=0.2)
    return binom.pmf(num_sites)

prob_sites_links_to(num_sites=3)

0.20132659200000017

In [61]:
ms.comb(10,3)*0.2**3*0.8**7

0.20132659200000011

### (b) $P(N_1 = 3 \text{ and } N_2 = 2)$

\begin{equation}
\begin{aligned}
    P(N_1 = 3 \text{ and } N_2 = 2) &= P(N_1 = 3)P(N_2 = 2) && \text{Independence assumption} \\
    &= 0.2013... \times {10 \choose 2}0.2^2(1-0.2)^{10-2} \\
    &\approx 0.061
\end{aligned}
\end{equation}

In [64]:
prob_sites_links_to(num_sites=3) * prob_sites_links_to(num_sites=2) 

0.060798594969501733

### (c) $Var(N_1)$

\begin{equation}
\begin{aligned}
    Var(N_1) &= np(1-p) && \text{Variance of Binomial Distribution} \\
    &= 10(0.2)(0.8) \\
    &= 1.6
\end{aligned}
\end{equation}

### (d) $Var(N_1 + N_2)$

\begin{equation}
\begin{aligned}
    Var(N_1 + N_2) &= Var(N_1) + Var(N_2) && \text{$N_1$ and $N_2$ are independent} \\
    &= 1.6+1.6 \\
    &= 3.2
\end{aligned}
\end{equation}

## Note: There are some good shortcuts in some of these problems, making the work much easier.
But you must JUSTIFY your work.

## 21. Let $X$ denote the number of heads we get by tossing a coin $50$ times. Consider Chebychev’s Inequality for the case of $2$ standard deviations. Compare the upper bound given by the inequality to the exact probability.

\begin{equation}
\begin{aligned}
    P(\left| X-\mu \right| \geq c \sigma) &\leq \frac{1}{c^2} && \text{Chebyshev's Inequality} \\
    P(\left| X-\mu \right| \geq 2 \sigma) &\leq \frac{1}{2^2} \\
    P(\left| X-\mu \right| \geq 2 \sqrt{\sigma^2}) &\leq \frac{1}{4} \\
    P(\left| X-\mu \right| \geq 2 \sqrt{np(1-p)}) &\leq && \text{Variance of binomial distribution} \\
    P(\left| X-\mu \right| \geq 2 \sqrt{50(0.5)(0.5)}) &\leq && \text{n=50, p=0.5} \\
    P(\left| X-\mu \right| \geq 5\sqrt{2}) &\leq \\
    P(\left| X-(np)\right| \geq 5\sqrt{2}) &\leq && \text{Expected Value of binomial distribution} \\
    P(X \geq 25 + 5\sqrt{2}) &\leq \\
    P(X \geq 25 + 7.07...) &\leq \\
    P(X \geq 32.07...) &\leq \\
    P(X \geq 33) &\leq \\
    0.016... &\leq
\end{aligned}
\end{equation}



In [68]:
def prob_geq_33_heads(num_tosses=50):
    def prob_coin_tosses(num_heads=33, num_tosses=50):
        binom = ss.binom(n=50, p=0.5)
        return binom.pmf(num_heads)
    
    total_prob = 0.0
    for i in range(33, num_tosses+1):
        total_prob += prob_coin_tosses(num_heads=i, num_tosses=num_tosses)
        
    return total_prob

In [69]:
prob_geq_33_heads(num_tosses=50)

0.016419568782134037

## 22. Suppose the number $N$ of cars arriving during a given time period at a toll booth has a Poisson distribution with parameter $\lambda$. Each car has a probability $p$ of being in a car pool. Let $M$ be the number of car-pool cars that arrive in the given period. Show that $M$ also has a Poisson distribution, with parameter $p\lambda$. (Hint: Use the Maclaurin series for $e^x$.)

\begin{equation}
\begin{aligned}
    P(X=k) &= \frac{e^{-\lambda}\lambda^k}{k!}, k=0,1,2,...
\end{aligned}
\end{equation}

\begin{equation}
\begin{aligned}
    e^\lambda = \sum_{k=0}^{\infty} \frac{\lambda^k}{k!}
\end{aligned}
\end{equation}

\begin{equation}
\begin{aligned}
    P(M=k) &= \sum_{n=0}^{\infty}P(M=k|N=n)P(N=n) \\
    &= \sum_{n=0}^{\infty}{n \choose k}p^k(1-p)^{n-k}\frac{e^{-\lambda}\lambda^n}{n!} \\
    &= \sum_{n=k}^{\infty}{n \choose k}p^k(1-p)^{n-k}\frac{e^{-\lambda}\lambda^n}{n!} && \text{n has to be at least k}\\
    &= \sum_{n=k}^{\infty}\frac{n!}{(n-k)!k!}p^k(1-p)^{n-k}\frac{e^{-\lambda}\lambda^n}{n!} \\
    &= \sum_{n=k}^{\infty}\frac{p^k(1-p)^{n-k}e^{-\lambda}\lambda^n}{(n-k)!k!} \\
    &= \sum_{n=k}^{\infty}\frac{p^k(1-p)^{n-k}e^{-\lambda}\lambda^k\lambda^{n-k}}{(n-k)!k!} \\
    &= \sum_{n=k}^{\infty}\frac{e^{-\lambda}p^k\lambda^k}{k!}\frac{(1-p)^{n-k}\lambda^{n-k}}{(n-k)!} \\
    &= \frac{e^{-\lambda}(p\lambda)^k}{k!}\sum_{n=k}^{\infty}\frac{((1-p)\lambda)^{n-k}}{(n-k)!}  && \text{Use Maclaurin Series for $e^x$}\\
    &= \frac{e^{-\lambda}(p\lambda)^k}{k!}e^{(1-p)\lambda} \\
    &= \frac{e^{-\lambda}(p\lambda)^k}{k!}e^\lambda e^{-\lambda p} \\
    &= \frac{e^{-(p\lambda)}(p\lambda)^k}{k!} \\
\end{aligned}
\end{equation}    

## 23. Consider a three-sided die, as on page 36. Let X denote the number of dots obtained in one roll.

## (a) State the value of $p_x(2):$
\begin{equation}
    \begin{aligned}
        p_x(2) &= P(X=2) \\
        &= \frac{1}{3}
    \end{aligned}
\end{equation}
    
## (b) Find $EX$ and $Var(X)$:

\begin{equation}
    \begin{aligned}
        EX &= 1(1/3) + 2(1/3) + 3(1/3) \\
        &= 1/3 + 2/3 + 3/3 \\
        &= 2
    \end{aligned}
\end{equation}

\begin{equation}
    \begin{aligned}
        Var(X) &= E(X^2) - E(X)^2 \\
        &= 1^2(1/3) + 2^2(1/3) + 3^2(1/3) - 4 \\
        &= 1/3 + 4(1/3) + 9(1/3) - 4 \\
        &= 1/3 + 4/3 + 9/3 - 4 \\
        &= 14/3 - 4 \\
        &= 2/3
    \end{aligned}
\end{equation}

## (c) Suppose you win $\$2$ for each dot. Find $EW$, where $W$ is the amount you win:

\begin{equation}
    \begin{aligned}
        W &= 2X \\
        EW &= E(2X) \\
        &= E(2)E(X) \\
        &= 2EX \\
        &= 2(2) \\
        &= 4
    \end{aligned}
\end{equation}



## 24. Consider the parking space problem in Section 4.2.2. Find $Var(M)$, where $M$ is the number of empty spaces in the first block, and $Var(D)$.

\begin{equation}
    \begin{aligned}
        Var(M) &= np(1-p) && \text{Variance of binomial distribution} \\
        &= 10(0.15)(1-0.15) \\
        &= 1.275
    \end{aligned}
\end{equation}

\begin{equation}
    \begin{aligned}
        Var(D) &= E(D^2) - E(D)^2 \\
        &= \left(\sum_{i=1}^{10}(11-i)^2(1-0.15)^{i-1}0.15 + \sum_{i=11}^{\infty}(i-11)^2(1-0.15)^{i-1}0.15\right) \\
        &\quad - 43.09366570787748 \\
        &= \sum_{i=1}^{10}(i-11)^2(1-0.15)^{i-1}0.15 + \sum_{i=11}^{\infty}(i-11)^2(1-0.15)^{i-1}0.15 \\
        &\quad - 43.09366570787748 && \text{$(i-11)^2$ and $(11-i)^2$ are the same}\\
        &= 0.15\sum_{i=1}^{\infty}(i-11)^2(1-0.15)^{i-1} - 43.09366570787748 \\
        &= 0.15\sum_{i=1}^{\infty}(i^2 - 22i + 121)(1-0.15)^{i-1} - 43.09366570787748 \\
        &= 0.15\sum_{i=1}^{\infty}(i^2)(1-0.15)^{i-1} - 0.15\sum_{i=1}^{\infty}(22i)(1-0.15)^{i-1} \\
        &\quad + 0.15\sum_{i=1}^{\infty}(121)(1-0.15)^{i-1} - 43.09366570787748 \\
        &= 0.15\sum_{i=1}^{\infty}(i)(i+1)(1-0.15)^{i-1} - 0.15\sum_{i=1}^{\infty}(i)(1-0.15)^{i-1} \\
        &\quad - 3.3\sum_{i=1}^{\infty}i(1-0.15)^{i-1} + 18.15\sum_{i=1}^{\infty}(1-0.15)^{i-1} - 43.09366570787748 \\
        &= 0.15\sum_{i=1}^{\infty}(i)(i+1)(1-0.15)^{i-1} - 3.45\sum_{i=1}^{\infty}i(1-0.15)^{i-1} \\
        &\quad + 18.15\sum_{i=1}^{\infty}(1-0.15)^{i-1} - 43.09366570787748 \\
        &= 0.15\sum_{i=1}^{\infty}(i)(i+1)(1-0.15)^{i-1} - 3.45\sum_{i=1}^{\infty}i(1-0.15)^{i-1} \\
        &\quad + 18.15\sum_{j=0}^{\infty}(1-0.15)^{j} - 43.09366570787748 && \text{Substitute $j=i-1$} \\
        &= 0.15\sum_{n=2}^{\infty}(n-1)(n)(1-0.15)^{n-2} - 3.45\sum_{i=1}^{\infty}i(1-0.15)^{i-1} \\
        &\quad + 18.15\sum_{j=0}^{\infty}(1-0.15)^{j} - 43.09366570787748 && \text{Substitute $n=i+1$} \\
        &= 0.15\left(\frac{2}{(1-0.85)^3}\right) - 3.45\left(\frac{1}{(1-0.85)^2}\right) \\
        &\quad + 18.15\left(\frac{1}{1-0.85}\right) - 43.09366570787748 \\
        &\approx 13.46
    \end{aligned} 
\end{equation}

In [102]:
# E(D)^2
(10.897909915861522 - 4.333333333333333)**2

43.09366570787748

In [133]:
# Var(H)
2/(0.15**2) -3.45/(0.15)**2 + 18.15/0.15 - (10.897909915861522 - 4.333333333333333)**2

13.461889847678066

In [105]:
    variance(park_sim(), 'num_open_spaces_first_block')

1.2754199974999998

In [135]:
variance(park_sim(num_experiments=10000000), 'dist_from_dest')

13.460245502952382

## 25. Suppose $X$ and $Y$ are independent, with variances $1$ and $2$, respectively. Find the value of $c$ that minimizes $Var[cX + (1-c)Y]$.

\begin{equation}
    \begin{aligned}
        Var[cX + (1-c)Y] &= c^2Var(X) + (1-c)^2Var(Y) \\
        &= c^2(1) + (1-c)^2(2) && \text{$c=1$ minimizes the above}\\
        &= (1)^2(1) + (1-1)^2(2) \\
        &= 1
    \end{aligned}
\end{equation}

## 26. In the cards example in Section 2.15.1, let $H$ denote the number of hearts. Find $EH$ and $Var(H)$.

\begin{equation}
    \begin{aligned}
        EH &= \sum_{i=0}^{4}P(H=i) \\
        &= 0P(H=0) + 1P(H=1) + 2P(H=2) + 3P(H=3) + 4P(H=4) \\
        &= 1\frac{{13 \choose 1}{39 \choose 4}}{{52 \choose 5}} + 2\frac{{13 \choose 2}{39 \choose 3}}{{52 \choose 5}} + 3\frac{{13 \choose 3}{39 \choose 2}}{{52 \choose 5}} + 4\frac{{13 \choose 4}{39 \choose 1}}{{52 \choose 5}} \\
        &\approx 1.25
    \end{aligned}
\end{equation}

In [95]:
# Expected Value
(1*ms.comb(13,1)*ms.comb(39,4) \
+ 2*ms.comb(13,2)*ms.comb(39,3) \
+ 3*ms.comb(13,3)*ms.comb(39,2) \
+ 4*ms.comb(13,4)*ms.comb(39,1)) / ms.comb(52,5)

1.2475240096038416

In [98]:
# E(H)^2
((1*ms.comb(13,1)*ms.comb(39,4) \
+ 2*ms.comb(13,2)*ms.comb(39,3) \
+ 3*ms.comb(13,3)*ms.comb(39,2) \
+ 4*ms.comb(13,4)*ms.comb(39,1)) / ms.comb(52,5))**2

1.5563161545380457

\begin{equation}
    \begin{aligned}
        Var(H) &= E(H^2) - E(H)^2 \\
        &= 0^2P(H=0) + 1^2P(H=1) + 2^2P(H=2) + 3^2P(H=3) + 4^2P(H=4) - E(H)^2\\
        &= 1\frac{{13 \choose 1}{39 \choose 4}}{{52 \choose 5}} + 4\frac{{13 \choose 2}{39 \choose 3}}{{52 \choose 5}} + 9\frac{{13 \choose 3}{39 \choose 2}}{{52 \choose 5}} + 16 \frac{{13 \choose 4}{39 \choose 1}}{{52 \choose 5}} - E(H)^2 \\
        &\approx 2.41409 - (1.2475)^2 \\
        &\approx 0.86
    \end{aligned}
\end{equation}

In [97]:
# E(H^2)
(1**2*ms.comb(13,1)*ms.comb(39,4) \
+ 2**2*ms.comb(13,2)*ms.comb(39,3) \
+ 3**2*ms.comb(13,3)*ms.comb(39,2) \
+ 4**2*ms.comb(13,4)*ms.comb(39,1)) / ms.comb(52,5)

2.4140906362545018

In [99]:
# Var(H)
2.4140906362545018 - 1.5563161545380457

0.8577744817164561

In [92]:
def draw_5_cards(num_experiments=10000):
    num_cards = 52
    num_hearts = 13
    num_others = num_cards-num_hearts
    heart_cards = np.ones(num_hearts)
    other_cards = np.zeros(num_others)
    all_cards = np.append(heart_cards, other_cards)
    
    num_hearts_results = []
    for i in range(num_experiments):
        num_hearts_results.append(np.random.permutation(all_cards)[0:5].sum())
        
    return pd.DataFrame({'num_hearts': num_hearts_results})

In [94]:
expected_value(draw_5_cards())

num_hearts    1.2617
dtype: float64

In [101]:
variance(draw_5_cards(), 'num_hearts')

0.86944123999999978

## 27. In the bank example in Section 4.5, suppose you observe the bank for $n$ days. Let $X$ denote the number of days in which at least $2$ customers entered during the 11:00-11:15 observation period. Find $P(X = k)$.

\begin{equation}
    \begin{aligned}
        P(Y=z) &= \frac{e^{−λ}λ^z}{z!}, z = 0,1,2,... && \text{$Y$ is # of customers who arrive between 11:00-11:15} \\
        P(Y\geq 2) &= \sum_{z=2}^{\infty}\frac{e^{−λ}λ^z}{z!} && \text{probability that there will be at least two customers entering during the time period} \\
        P(X=k) &= {n \choose k}p^k(1-p)^{n-k} && \text{Assuming that days are independent from one another, $P(X=k)$ is binomially distributed} \\
        &= {n \choose k}P(Y\geq2)^kP(Y<2)^{n-k} \\
        &= {n \choose k}\left(\sum_{z=2}^{\infty}\frac{e^{−λ}λ^z}{z!}\right)^k\left(\sum_{z=0}^{1}\frac{e^{−λ}λ^z}{z!}\right)^{n-k} \\
    \end{aligned}
\end{equation}


## 28. Find $E(X^3)$, where $X$ has a geometric distribution with parameter $p$.

\begin{equation}
    \begin{aligned}
        E(X^3) &= \sum_{k=1}^{\infty}k^3(1-p)^{k-1}p \\
        &= p\sum_{k=1}^{\infty}k(k+1)(k+2)(1-p)^{k-1} - p\sum_{k=1}^{\infty}(3k^2+2k)(1-p)^{k-1} \\
        &= p\sum_{j=2}^{\infty}j(j-1)(j-2)(1-p)^{j-3} - p\sum_{k=1}^{\infty}(3k^2+2k)(1-p)^{k-1} && \text{Substitute $j=k+3$ for first sum} \\
        &= \frac{6p}{(1-(1-p))^4} - p\sum_{k=1}^{\infty}(3k^2+2k)(1-p)^{k-1} && \text{Take derivative 3x of $\sum_{j=0}^\infty (q)^j = \frac{1}{1-(q)}$} \\
        &= \frac{6}{p^3} - p\sum_{k=1}^{\infty}(3k^2+2k)(1-p)^{k-1} \\
        &= \frac{6}{p^3} - 3p\sum_{k=1}^{\infty}k^2(1-p)^{k-1} - 2p\sum_{k=1}^{\infty}k(1-p)^{k-1}\\
        &= \frac{6}{p^3} - 3p\sum_{k=1}^{\infty}(k^2+k)(1-p)^{k-1} + 3p\sum_{k=1}^{\infty}k(1-p)^{k-1} - 2p\sum_{k=1}^{\infty}k(1-p)^{k-1}\\
        &= \frac{6}{p^3} - 3p\sum_{k=1}^{\infty}(k^2+k)(1-p)^{k-1} + p\sum_{k=1}^{\infty}k(1-p)^{k-1} \\
        &= \frac{6}{p^3} - 3p\sum_{k=1}^{\infty}(k^2+k)(1-p)^{k-1} + \frac{p}{(1-(1-p))^2} && \text{Take derivative of $\sum_{j=0}^\infty (q)^j = \frac{1}{1-(q)}$}\\
        &= \frac{6}{p^3} - 3p\sum_{k=1}^{\infty}(k^2+k)(1-p)^{k-1} + \frac{1}{p} \\
        &= \frac{6}{p^3} - 3p\sum_{k=1}^{\infty}k(k+1)(1-p)^{k-1} + \frac{1}{p} \\
        &= \frac{6}{p^3} - 3p\sum_{j=2}^{\infty}(j-1)(j)(1-p)^{j-2} + \frac{1}{p} && \text{Substitute $j=k+1$}\\
        &= \frac{6}{p^3} - 3p\left(\frac{2}{(1-(1-p))^3}\right)+ \frac{1}{p} && \text{Take derivative 2x of $\sum_{j=0}^\infty (q)^j = \frac{1}{1-(q)}$} \\
        &= \frac{6}{p^3} - \frac{6}{p^2} + \frac{1}{p} \\
    \end{aligned}
\end{equation}



## 29. Supppose we have a nonnegative random variable $X$, and define a new random variable $Y$, which is equal to $X$ if $X > 8$ and equal to $0$ otherwise. Assume $X$ takes on only a finite number of values (just a mathematical nicety, not really an issue). Which one of the following is true:

### (i) $EY \leq EX$. 

### (ii) $EY ≥ EX$.

### (iii) Either of $EY$ and $EX$ could be larger than the other, depending on the situation. 

### (iv) $EY$ is undefined.

\begin{equation}
    \begin{aligned}
        EY &= \sum_{j=0}^8 0P(X=j) + \sum_{j=9}^{\infty} jP(X=j) \\
           &= \sum_{j=9}^{\infty} jP(X=j) \\
        EX &= \sum_{j=0}^{\infty} jP(X=j) \\
        EX - EY &= \sum_{j=0}^{\infty} jP(X=j) - \sum_{j=9}^{\infty} jP(X=j)\\
        &= \sum_{j=0}^{8} jP(X=j) \\
        &\geq 0 \\
        EY &\leq EX 
    \end{aligned}
\end{equation}

## 30. Say we roll two dice, a blue one and a yellow one. Let $B$ and $Y$ denote the number of dots we get, respectively, and write $S = B + Y$. Now let $G$ denote the indicator random variable for the event $S = 2$. Find $E(G)$.

\begin{equation}
    \begin{aligned}
        EG &= 1P(S=2) + 0P(S\neq 2) \\
        &= P(S=2) \\
        &= P(B=1, Y=1) \\
        &= \frac{1}{6} \times \frac{1}{6} \\
        &= \frac{1}{36} \\
    \end{aligned}
\end{equation}

## 32. Consider the bus ridership example, Section 2.12. Suppose upon arrival to a certain stop, there are $2$ passengers. Let $A$ denote the number of them who choose to alight at that stop.

###(a) State the parametric family that the distribution of $A$ belongs to.

Binomial distribution with $n=2$.

###(b) Find $p_A(1)$ and $F_A(1)$, writing each answer in decimal expression form e.g. $128 \times 0.32+0.3333$.

\begin{equation}
    \begin{aligned}
        p_A(1) &= P(A=1) \\
        &= {2 \choose 1}p(1-p) \\
        &= 2(0.2)(0.8) \\
        &= 0.32
    \end{aligned}
\end{equation}

Not really sure what $F_A(1)$ stands for.

### 33. Suppose you have a large disk farm, so heavily used that the lifetimes $L$ are measured in months. They come from two different factories, in proportions $q$ and $1-q$. The disks from factory $i$ have geometrically distributed lifetime with parameter $p_i$, $i = 1,2$. Find $Var(L)$ in terms of $q$ and the $p_i$.

\begin{equation}
    \begin{aligned}
        L &= qL_1 + (1-q)L_2 \\
        EL &= E(qL_1 + (1-q)L_2) \\
        &= qE(L_1) + (1-q)E(L_2) \\
        &= q\sum_{k=0}^{\infty}k(1-p_1)^{k-1} + (1-q)\sum_{k=0}^{\infty}k(1-p_2)^{k-1}  \\
        &= qp_1\left(\frac{1}{(1-(1-p_1))^2}\right) + (1-q)p_2\left(\frac{1}{(1-(1-p_2))^2}\right) \\
        &= \frac{q}{p_1} + \frac{1-q}{p_2} \\
        E(L)^2 &= \left(\frac{q}{p_1} + \frac{1-q}{p_2}\right)^2 \\
        E(L^2) &= qE(L_1^2) + (1-q)E(L_2^2) \\
        &= qp_1\sum_{k=1}^{\infty}k^2(1-p_1)^{k-1} + (1-q)p_2\sum_{j=1}^{\infty}j^2(1-p_2)^{j-1} \\
        &= qp_1\left(\sum_{k=1}^{\infty}k(k+1)(1-p_1)^{k-1} + \sum_{k=1}^{\infty}k(1-p_1)^{k-1}\right) \\
        &\quad + (1-q)p_2\left(\sum_{j=1}^{\infty}j(j+1)(1-p_2)^{j-1} - \sum_{j=1}^{\infty}j(1-p_2)^{j-1}\right) \\
        &= qp_1\left(\sum_{m=2}^{\infty}m(m-1)(1-p_1)^{m-2} + \sum_{k=1}^{\infty}k(1-p_1)^{k-1}\right) \\
        &\quad + (1-q)p_2\left(\sum_{n=2}^{\infty}n(n-1)(1-p_2)^{n-2} - \sum_{j=1}^{\infty}j(1-p_2)^{j-1}\right) \\
        &= qp_1\left(\frac{2}{(1-(1-p_1))^3}-\frac{1}{(1-(1-p_1))^2}\right) \\ 
        &\quad + (1-q)p_2\left(\frac{2}{(1-(1-p_2))^3}-\frac{1}{(1-(1-p_2))^2}\right) \\
        &= q\left(\frac{2}{p_1^2}-\frac{1}{p_1}\right) + (1-q)\left(\frac{2}{p_2^2}-\frac{1}{p_2}\right) \\
        Var(L) &= E(L^2) - E(L)^2 \\
        &=  q\left(\frac{2}{p_1^2}-\frac{1}{p_1}\right) + (1-q)\left(\frac{2}{p_2^2}-\frac{1}{p_2}\right) \\
        &\quad - \left(\frac{q}{p_1} + \frac{1-q}{p_2}\right)^2 \\
    \end{aligned}
\end{equation}


In [265]:
def disk_farm(num_experiments=10000000,
              from_factory_1_prob=0.3,
              factory_1_disk_failure_prob=0.2,
              factory_2_disk_failure_prob=0.4):
    fac_1_lifetimes = np.random.geometric(p=factory_1_disk_failure_prob,
                        size=num_experiments)
    fac_2_lifetimes = np.random.geometric(p=factory_2_disk_failure_prob,
                        size=num_experiments)
    
    return np.append(fac_1_lifetimes[0:int(num_experiments*from_factory_1_prob)],
                     fac_2_lifetimes[int(num_experiments*from_factory_1_prob):])



In [293]:
disk_farm(from_factory_1_prob=0.5,
          factory_1_disk_failure_prob=0.01,
          factory_2_disk_failure_prob=0.01).mean()

100.029647

In [294]:
def expected_value_disk_farm(from_factory_1_prob=0.5,
          factory_1_disk_failure_prob=0.01,
          factory_2_disk_failure_prob=0.01):
    return from_factory_1_prob/factory_1_disk_failure_prob \
    + (1-from_factory_1_prob)/factory_2_disk_failure_prob

expected_value_disk_farm(from_factory_1_prob=0.5,
      factory_1_disk_failure_prob=0.01,
      factory_2_disk_failure_prob=0.01)

100.0

In [341]:
# Issue: This is the first attempt. I think this does not work because
# it assumes that covariance between the lifetime of items from factory
# 1 is zero (i.e. independent of the lifetimes from factory 2).
# But aren't they?
def variance_disk_farm(from_factory_1_prob=0.3,
                       factory_1_disk_failure_prob=0.2,
                       factory_2_disk_failure_prob=0.4):
    # (q^2)*(1-p_1)/p_1^2 + (1-q)^2*(1-p_2)/p_2^2
    
    return from_factory_1_prob**2 * (1-factory_1_disk_failure_prob) \
    / factory_1_disk_failure_prob**2 \
    + (1-factory_2_disk_failure_prob) / factory_2_disk_failure_prob**2 \
    - (from_factory_1_prob)**2 * (1-factory_2_disk_failure_prob) \
    / factory_2_disk_failure_prob**2
#     return from_factory_1_prob**2 \
#             * (1-factory_1_disk_failure_prob) \
#             / factory_1_disk_failure_prob**2 \
#             + (1-from_factory_1_prob)**2 \
#             * (1-factory_2_disk_failure_prob) \
#             / factory_2_disk_failure_prob**2
          


In [342]:
# This is the implementation of my answer above. Seems to give a very good approximation 
# of the simulated answer.
def variance_disk_farm_2nd_version(q=0.2,
                       p_1=0.09,
                       p_2=0.01):
    return q*(2/p_1**2 - 1/p_1) + (1-q)*(2/p_2**2-1/p_2) - (q/p_1+(1-q)/p_2)**2

In [334]:
variance_disk_farm_2nd_version(q=0.5,
                       p_1=0.01,
                       p_2=0.01)

9900.0

In [337]:
variance_disk_farm_2nd_version(q=0.2,
                       p_1=0.09,
                       p_2=0.01)

9206.666666666666

In [323]:
disk_farm(from_factory_1_prob=0.2,
          factory_1_disk_failure_prob=0.09,
          factory_2_disk_failure_prob=0.01,
          num_experiments=100000000).var()

9206.205916455734